diff --git a/browser/base/jar.mn b/browser/base/jar.mn
index 26e70f4f410443f4cdd28b6a8afc65a0dfc5f183..186baf6b8cda8d2cd0c37e27f94d18b911164b88 100644
--- a/browser/base/jar.mn
+++ b/browser/base/jar.mn
@@ -77,11 +77,6 @@ browser.jar:
         content/browser/defaultthemes/5.icon.jpg            (content/defaultthemes/5.icon.jpg)
         content/browser/defaultthemes/5.preview.jpg         (content/defaultthemes/5.preview.jpg)
         content/browser/history-swipe-arrow.svg             (content/history-swipe-arrow.svg)
-*       content/browser/pageinfo/pageInfo.xhtml             (content/pageinfo/pageInfo.xhtml)
-        content/browser/pageinfo/pageInfo.js                (content/pageinfo/pageInfo.js)
-        content/browser/pageinfo/pageInfo.css               (content/pageinfo/pageInfo.css)
-        content/browser/pageinfo/permissions.js             (content/pageinfo/permissions.js)
-        content/browser/pageinfo/security.js                (content/pageinfo/security.js)
         content/browser/robot.ico                           (content/robot.ico)
         content/browser/static-robot.png                    (content/static-robot.png)
         content/browser/safeMode.css                        (content/safeMode.css)
