diff --git a/browser/base/content/browser-siteIdentity.js b/browser/base/content/browser-siteIdentity.js
index 01d47f0907e6b77423ae4ad00150d71e31eb8ea8..fa442e59c4787b31c210d4e6c4c44b2ff67f0f1a 100644
--- a/browser/base/content/browser-siteIdentity.js
+++ b/browser/base/content/browser-siteIdentity.js
@@ -57,7 +57,7 @@ var gIdentityHandler = {
    * RegExp used to decide if an about url should be shown as being part of
    * the browser UI.
    */
-  _secureInternalPages: /^(?:accounts|addons|cache|certificate|config|crashes|downloads|license|logins|preferences|protections|rights|sessionrestore|support|welcomeback|ion)(?:[?#]|$)/i,
+  _secureInternalPages: /^(?:accounts|addons|cache|certificate|config|crashes|downloads|license|logins|newpreferences|preferences|protections|rights|sessionrestore|support|welcomeback|ion|history)(?:[?#]|$)/i,
 
   /**
    * Whether the established HTTPS connection is considered "broken".
@@ -729,6 +729,15 @@ var gIdentityHandler = {
     return "unknownIdentity";
   },
 
+  /**
+   * Returns the reader mode button element.
+  */
+  get _readerModeButton() {
+    return (this._readerModeButton = document.getElementById(
+      "reader-mode-button"
+    ));
+  },
+
   /**
    * Returns whether the issuer of the current certificate chain is
    * built-in (returns false) or imported (returns true).
@@ -770,12 +779,20 @@ var gIdentityHandler = {
       icon_label = brandBundle.getString("brandShorterName");
     } else if (this._pageExtensionPolicy) {
       // This is a WebExtension page.
-      this._identityBox.className = "extensionPage";
-      let extensionName = this._pageExtensionPolicy.name;
-      icon_label = gNavigatorBundle.getFormattedString(
-        "identity.extension.label",
-        [extensionName]
-      );
+      let displayUri = this._uri.displaySpec || null;
+
+      if (BROWSER_NEW_TAB_URL == displayUri) {
+        this._identityIconLabel.setAttribute("hidden", "true");
+        this._identityBox.setAttribute("pageproxystate", "invalid");
+        gProtectionsHandler._trackingProtectionIconContainer.setAttribute("hidden", true);
+      } else {
+        this._identityBox.className = "extensionPage";
+        let extensionName = this._pageExtensionPolicy.name;
+        icon_label = gNavigatorBundle.getFormattedString(
+          "identity.extension.label",
+          [extensionName]
+        );
+      }
     } else if (this._uriHasHost && this._isSecureConnection) {
       // This is a secure connection.
       this._identityBox.className = "verifiedDomain";
@@ -816,6 +833,12 @@ var gIdentityHandler = {
     } else if (this._isPotentiallyTrustworthy) {
       // This is a local resource (and shouldn't be marked insecure).
       this._identityBox.className = "localResource";
+
+      let readerActive = this._readerModeButton.getAttribute("readeractive");
+
+      if (readerActive == "true") {
+        this._identityBox.className = "readerIdentity";
+      }
     } else {
       // This is an insecure connection.
       let warnOnInsecure =
