diff --git a/browser/themes/shared/jar.inc.mn b/browser/themes/shared/jar.inc.mn
index 9a62864b3f1ff80db532019949f6c480175de35f..88ec22dc5653965b13401b07e695a524de501bf4 100644
--- a/browser/themes/shared/jar.inc.mn
+++ b/browser/themes/shared/jar.inc.mn
@@ -203,6 +203,7 @@
 #ifndef MOZ_WIDGET_GTK
   skin/classic/browser/sort.svg                       (../shared/icons/sort.svg)
 #endif
+  skin/classic/browser/spell-check.svg                (../shared/icons/spell-check.svg)
   skin/classic/browser/stop.svg                       (../shared/icons/stop.svg)
   skin/classic/browser/stop-to-reload.svg             (../shared/icons/stop-to-reload.svg)
   skin/classic/browser/success-animation.svg          (../shared/icons/success-animation.svg)
