diff --git a/browser/components/urlbar/UrlbarInput.jsm b/browser/components/urlbar/UrlbarInput.jsm
index 4c4207106b38e9da1a2ab1f2ce6c0a6d71d2ad6b..ca675d8f9d3ad309161878edc9a9a8157bbc5751 100644
--- a/browser/components/urlbar/UrlbarInput.jsm
+++ b/browser/components/urlbar/UrlbarInput.jsm
@@ -2538,6 +2538,8 @@ class UrlbarInput {
       .GetStringFromName("pasteAndGo.label");
     pasteAndGo.setAttribute("label", label);
     pasteAndGo.setAttribute("anonid", "paste-and-go");
+    pasteAndGo.setAttribute("image", "chrome://global/skin/icons/search-glass.svg");
+    pasteAndGo.classList.add("menuitem-iconic");
     pasteAndGo.addEventListener("command", () => {
       this._suppressStartQuery = true;
 
