diff --git a/browser/moz.build b/browser/moz.build
index ca88ad0a94f7339621e236d64776a3b306371aa4..31740cccf402661f13f5d25746a846783ba4752f 100644
--- a/browser/moz.build
+++ b/browser/moz.build
@@ -49,6 +49,10 @@ if CONFIG["MOZ_GPSD"]:
 
 # These files are specified in this moz.build to pick up DIST_SUBDIR as set in
 # this directory, which is un-set in browser/app.
+JS_PREFERENCE_PP_FILES += [
+    "app/profile/dot.js",
+]
+
 JS_PREFERENCE_PP_FILES += [
     "app/profile/firefox.js",
 ]
