diff --git a/browser/components/extensions/parent/ext-sidebarAction.js b/browser/components/extensions/parent/ext-sidebarAction.js
index 99aa63874bc6f1821baae6788fbba18f6861f9c6..9fe8b1800715f5e33d647bfe23a1058f56ea2900 100644
--- a/browser/components/extensions/parent/ext-sidebarAction.js
+++ b/browser/components/extensions/parent/ext-sidebarAction.js
@@ -178,7 +178,7 @@ this.sidebarAction = class extends ExtensionAPI {
     menuitem.setAttribute("type", "checkbox");
     menuitem.setAttribute("label", details.title);
     menuitem.setAttribute("oncommand", `SidebarUI.toggle("${this.id}");`);
-    menuitem.setAttribute("class", "menuitem-iconic webextension-menuitem");
+    menuitem.setAttribute("class", "sidebar-iconbar-item-ext sidebar-iconbar-item");
     menuitem.setAttribute("key", keyId);
     this.setMenuIcon(menuitem, details);
 
@@ -195,8 +195,8 @@ this.sidebarAction = class extends ExtensionAPI {
     this.setMenuIcon(toolbarbutton, details);
 
     document.getElementById("viewSidebarMenu").appendChild(menuitem);
-    let separator = document.getElementById("sidebar-extensions-separator");
-    separator.parentNode.insertBefore(toolbarbutton, separator);
+    let iconBar = document.getElementById("sidebar-iconbar");
+    iconBar.childNodes[0].appendChild(toolbarbutton);
     SidebarUI.updateShortcut({ button: toolbarbutton });
 
     return menuitem;
