diff --git a/browser/components/customizableui/CustomizableUI.jsm b/browser/components/customizableui/CustomizableUI.jsm
index 5188ac328fbdbe4c7bab26349be2b43aaf1ad2e6..e5a9cf4154b0cdf8ff0868701d8163c6237ae32a 100644
--- a/browser/components/customizableui/CustomizableUI.jsm
+++ b/browser/components/customizableui/CustomizableUI.jsm
@@ -9,6 +9,9 @@ const { Services } = ChromeUtils.import("resource://gre/modules/Services.jsm");
 const { XPCOMUtils } = ChromeUtils.import(
   "resource://gre/modules/XPCOMUtils.jsm"
 );
+const { ExtensionCommon } = ChromeUtils.import(
+  "resource://gre/modules/ExtensionCommon.jsm"
+);
 const { AppConstants } = ChromeUtils.import(
   "resource://gre/modules/AppConstants.jsm"
 );
@@ -52,7 +55,10 @@ XPCOMUtils.defineLazyPreferenceGetter(
   false
 );
 
-const kDefaultThemeID = "default-theme@mozilla.org";
+const { makeWidgetId } = ExtensionCommon;
+
+const kDefaultThemeID = "dynamic@themes.dothq.co";
+const kShieldID = "shield@privacy.dothq.co";
 
 const kSpecialWidgetPfx = "customizableui-special-";
 
@@ -262,6 +268,7 @@ var CustomizableUIInternal = {
         ? null
         : "home-button",
       "spring",
+      `${makeWidgetId(kShieldID)}-browser-action`,
       "urlbar-container",
       "spring",
       "downloads-button",
