diff --git a/browser/modules/PageActions.jsm b/browser/modules/PageActions.jsm
index 3e26f72e7d3296f94bfe7ee85c8102e3c582e5c5..ff6abfa38d06236b8ede46c20e7c00a5572c8b55 100644
--- a/browser/modules/PageActions.jsm
+++ b/browser/modules/PageActions.jsm
@@ -1220,7 +1220,7 @@ Action.prototype = {
   },
 
   get _isBuiltIn() {
-    let builtInIDs = ["screenshots_mozilla_org"].concat(
+    let builtInIDs = ["screenshots_mozilla_org", "qr_dothq_co"].concat(
       gBuiltInActions.filter(a => !a.__isSeparator).map(a => a.id)
     );
     return builtInIDs.includes(this.id);
