diff --git a/browser/modules/ThemeVariableMap.jsm b/browser/modules/ThemeVariableMap.jsm
index a906416cc17f59d4a145f3f1e283114359e46489..bc60782c9292dee266478917deba03b180e76f78 100644
--- a/browser/modules/ThemeVariableMap.jsm
+++ b/browser/modules/ThemeVariableMap.jsm
@@ -173,6 +173,13 @@ const ThemeVariableMap = [
       optionalElementID: "browser",
     },
   ],
+  [
+    "--sidebar-highlight-color",
+    {
+      lwtProperty: "sidebar_highlight",
+      optionalElementID: "browser",
+    },
+  ],
 ];
 
 const ThemeContentPropertyList = [
