diff --git a/browser/base/content/browser-context.inc b/browser/base/content/browser-context.inc
index 2b474f879e49d5086bf0725864319be55de83956..8e460fb20290aa29f144a9f6bccc9257931ede42 100644
--- a/browser/base/content/browser-context.inc
+++ b/browser/base/content/browser-context.inc
@@ -224,20 +224,33 @@
       <menuseparator id="context-sep-ctp"/>
       <menuitem id="context-savepage"
                 data-l10n-id="main-context-menu-page-save"
-                oncommand="gContextMenu.savePageAs();"/>
+                oncommand="gContextMenu.savePageAs();"
+                class="menuitem-iconic"
+                image="chrome://browser/skin/save.svg"/>
+      <menuitem id="context-generateqrcode"
+                data-l10n-id="main-context-menu-page-qr-generate"
+                oncommand="gContextMenu.generateQrCode(event);"
+                class="menuitem-iconic"
+                image="chrome://browser/skin/qr-code.svg"/>
       <menuitem id="context-pocket"
                 data-l10n-id="main-context-menu-save-to-pocket"
                 oncommand="Pocket.savePage(gContextMenu.browser, gContextMenu.browser.currentURI.spec, gContextMenu.browser.contentTitle);"/>
       <menuseparator id="context-sep-sendpagetodevice" class="sync-ui-item"
                      hidden="true"/>
       <menu id="context-sendpagetodevice"
-            class="sync-ui-item"
+            class="sync-ui-item menuitem-iconic"
             data-l10n-id="main-context-menu-send-to-device"
-            hidden="true">
+            hidden="true"
+            image="chrome://browser/skin/send-to-device.svg">
         <menupopup id="context-sendpagetodevice-popup"
                    onpopupshowing="(() => { gSync.populateSendTabToDevicesMenu(event.target, gBrowser.currentURI.spec, gBrowser.contentTitle); })()"/>
       </menu>
-      <menu id="fill-login" hidden="true">
+      <menu
+        id="fill-login"
+        hidden="true"
+        class="menuitem-iconic"
+        image="chrome://browser/skin/login.svg"
+      >
         <menupopup id="fill-login-popup" />
       </menu>
      <menuitem id="fill-login-generated-password"
@@ -251,23 +264,33 @@
       <menuseparator id="passwordmgr-items-separator"/>
       <menuitem id="context-undo"
                 data-l10n-id="text-action-undo"
-                command="cmd_undo"/>
+                command="cmd_undo"
+                class="menuitem-iconic"
+                image="chrome://global/skin/icons/undo.svg"/>
       <menuitem id="context-redo"
                 data-l10n-id="text-action-redo"
                 command="cmd_redo"/>
       <menuseparator id="context-sep-redo"/>
       <menuitem id="context-cut"
                 data-l10n-id="text-action-cut"
-                command="cmd_cut"/>
+                command="cmd_cut"
+                class="menuitem-iconic"
+                image="chrome://browser/skin/edit-cut.svg"/>
       <menuitem id="context-copy"
                 data-l10n-id="text-action-copy"
-                command="cmd_copy"/>
+                command="cmd_copy"
+                class="menuitem-iconic"
+                image="chrome://browser/skin/edit-copy.svg"/>
       <menuitem id="context-paste"
                 data-l10n-id="text-action-paste"
-                command="cmd_paste"/>
+                command="cmd_paste"
+                class="menuitem-iconic"
+                image="chrome://browser/skin/edit-paste.svg"/>
       <menuitem id="context-delete"
                 data-l10n-id="text-action-delete"
-                command="cmd_delete"/>
+                command="cmd_delete"
+                class="menuitem-iconic"
+                image="chrome://global/skin/icons/delete.svg"/>
       <menuitem id="context-selectall"
                 data-l10n-id="text-action-select-all"
                 command="cmd_selectAll"/>
@@ -289,7 +312,9 @@
                 oncommand="AddKeywordForSearchField();"/>
       <menuitem id="context-searchselect"
                 oncommand="BrowserSearch.loadSearchFromContext(this.searchTerms, this.usePrivate, this.principal, this.csp, event);"
-                onclick="checkForMiddleClick(this, event);"/>
+                onclick="checkForMiddleClick(this, event);"
+                class="menuitem-iconic"
+                image="chrome://global/skin/icons/search-glass.svg"/>
       <menuitem id="context-searchselect-private"
                 oncommand="BrowserSearch.loadSearchFromContext(this.searchTerms, true, this.principal, this.csp, event);"
                 onclick="checkForMiddleClick(this, event);"/>
@@ -344,7 +369,9 @@
                 label="&spellCheckToggle.label;"
                 type="checkbox"
                 accesskey="&spellCheckToggle.accesskey;"
-                oncommand="InlineSpellCheckerUI.toggleEnabled(window);"/>
+                oncommand="InlineSpellCheckerUI.toggleEnabled(window);"
+                class="menuitem-iconic"
+                image="chrome://browser/skin/spell-check.svg"/>
       <menuitem id="spell-add-dictionaries-main"
                 label="&spellAddDictionaries.label;"
                 accesskey="&spellAddDictionaries.accesskey;"
@@ -377,11 +404,15 @@
       <menuitem id="context-inspect-a11y"
                 hidden="true"
                 data-l10n-id="main-context-menu-inspect-a11y-properties"
-                oncommand="gContextMenu.inspectA11Y();"/>
+                oncommand="gContextMenu.inspectA11Y();"
+                class="menuitem-iconic"
+                image="chrome://devtools/skin/images/command-pick-accessibility.svg"/>
       <menuitem id="context-inspect"
                 hidden="true"
                 data-l10n-id="main-context-menu-inspect"
-                oncommand="gContextMenu.inspectNode();"/>
+                oncommand="gContextMenu.inspectNode();"
+                class="menuitem-iconic"
+                image="chrome://devtools/skin/images/command-pick.svg"/>
       <menuseparator id="context-media-eme-separator" hidden="true"/>
       <menuitem id="context-media-eme-learnmore"
                 class="menuitem-iconic"
