diff --git a/browser/themes/shared/jar.inc.mn b/browser/themes/shared/jar.inc.mn
index ada11562915d4c63b5aedb8e0ed2b49512f2c5d5..6d4e4f889e1a87e69ddeb6345d8c05dc277bd3db 100644
--- a/browser/themes/shared/jar.inc.mn
+++ b/browser/themes/shared/jar.inc.mn
@@ -45,6 +45,9 @@
   skin/classic/browser/customizableui/density-touch.svg        (../shared/customizableui/density-touch.svg)
   skin/classic/browser/customizableui/menu-arrow.svg           (../shared/customizableui/menu-arrow.svg)
   skin/classic/browser/customizableui/whimsy.png               (../shared/customizableui/whimsy.png)
+  skin/classic/browser/dotui/sidebar.css                       (../shared/dotui/sidebar.css)
+  skin/classic/browser/dotui/webui/common.css                  (../shared/dotui/webui/common.css)
+* skin/classic/browser/dotui/webui/settings.css                  (../shared/dotui/webui/settings.css)
   skin/classic/browser/downloads/contentAreaDownloadsView.css  (../shared/downloads/contentAreaDownloadsView.css)
   skin/classic/browser/downloads/download-blocked.svg          (../shared/downloads/download-blocked.svg)
   skin/classic/browser/downloads/download-summary.svg          (../shared/downloads/download-summary.svg)
@@ -77,6 +80,8 @@
   skin/classic/browser/notification-icons/drag-indicator.svg                (../shared/notification-icons/drag-indicator.svg)
   skin/classic/browser/notification-icons/focus-tab-by-prompt.svg           (../shared/notification-icons/focus-tab-by-prompt.svg)
   skin/classic/browser/notification-icons/indexedDB.svg                     (../shared/notification-icons/indexedDB.svg)
+  skin/classic/browser/notification-icons/javascript.svg                    (../shared/notification-icons/javascript.svg)
+  skin/classic/browser/notification-icons/location.svg                      (../shared/notification-icons/location.svg)
   skin/classic/browser/notification-icons/microphone-blocked.svg            (../shared/notification-icons/microphone-blocked.svg)
   skin/classic/browser/notification-icons/microphone-detailed.svg           (../shared/notification-icons/microphone-detailed.svg)
   skin/classic/browser/notification-icons/microphone.svg                    (../shared/notification-icons/microphone.svg)
@@ -100,6 +105,10 @@
   skin/classic/browser/notification-icons/block-social-dark.svg             (../shared/notification-icons/block-social-dark.svg)
   skin/classic/browser/notification-icons/block-social.svg                  (../shared/notification-icons/block-social.svg)
 
+  skin/classic/browser/onboarding-icons/adblock.svg                         (../shared/onboarding-icons/adblock.svg)
+  skin/classic/browser/onboarding-icons/import.svg                          (../shared/onboarding-icons/import.svg)
+  skin/classic/browser/onboarding-icons/personalise.svg                     (../shared/onboarding-icons/personalise.svg)
+
   skin/classic/browser/tracking-protection.svg                 (../shared/identity-block/tracking-protection.svg)
   skin/classic/browser/tracking-protection-disabled.svg        (../shared/identity-block/tracking-protection-disabled.svg)
   skin/classic/browser/tracking-protection-animation.svg        (../shared/identity-block/tracking-protection-animation.svg)
@@ -163,9 +172,11 @@
   skin/classic/browser/bookmark-hollow.svg            (../shared/icons/bookmark-hollow.svg)
   skin/classic/browser/bookmark-star-on-tray.svg      (../shared/icons/bookmark-star-on-tray.svg)
   skin/classic/browser/bookmarks-toolbar.svg          (../shared/icons/bookmarks-toolbar.svg)
+  skin/classic/browser/certificate.svg                (../shared/icons/certificate.svg)
   skin/classic/browser/characterEncoding.svg          (../shared/icons/characterEncoding.svg)
   skin/classic/browser/chevron-animation.svg          (../shared/icons/chevron-animation.svg)
   skin/classic/browser/check-animation.svg            (../shared/icons/check-animation.svg)
+  skin/classic/browser/cookies.svg                    (../shared/icons/cookies.svg)
   skin/classic/browser/customize.svg                  (../shared/icons/customize.svg)
   skin/classic/browser/developer.svg                  (../shared/icons/developer.svg)
   skin/classic/browser/device-phone.svg               (../shared/icons/device-phone.svg)
@@ -206,6 +217,7 @@
   skin/classic/browser/private-browsing.svg           (../shared/icons/private-browsing.svg)
   skin/classic/browser/privateBrowsing.svg            (../shared/icons/privateBrowsing.svg)
   skin/classic/browser/restore-session.svg            (../shared/icons/restore-session.svg)
+  skin/classic/browser/qr-code.svg                    (../shared/icons/qr-code.svg)
   skin/classic/browser/quit.svg                       (../shared/icons/quit.svg)
   skin/classic/browser/reload.svg                     (../shared/icons/reload.svg)
   skin/classic/browser/reload-to-stop.svg             (../shared/icons/reload-to-stop.svg)
@@ -218,6 +230,7 @@
 #ifndef MOZ_WIDGET_GTK
   skin/classic/browser/sort.svg                       (../shared/icons/sort.svg)
 #endif
+  skin/classic/browser/spell-check.svg                (../shared/icons/spell-check.svg)
   skin/classic/browser/stop.svg                       (../shared/icons/stop.svg)
   skin/classic/browser/stop-to-reload.svg             (../shared/icons/stop-to-reload.svg)
   skin/classic/browser/sync.svg                       (../shared/icons/sync.svg)
