diff --git a/browser/components/controlcenter/content/identityPanel.inc.xhtml b/browser/components/controlcenter/content/identityPanel.inc.xhtml
index 8875b2ac8d0bf878fee7a365d07553bc1be5e653..79249d3458b19c457c9c63549ae89418b1c95551 100644
--- a/browser/components/controlcenter/content/identityPanel.inc.xhtml
+++ b/browser/components/controlcenter/content/identityPanel.inc.xhtml
@@ -23,12 +23,14 @@
           <html:span id="identity-popup-mainView-panel-header-span" role="heading" aria-level="1"/>
         </label>
       </vbox>
+      
       <toolbarseparator></toolbarseparator>
 
       <!-- Security Section -->
-      <hbox id="identity-popup-security" class="identity-popup-section" flex="1">
-        <vbox class="identity-popup-security-content" flex="1">
+      <hbox id="identity-popup-security" class="identity-popup-section" flex="1" style="display: none;">
+        <vbox class="identity-popup-security-content identity-popup-section-content" flex="1">
 
+          <!-- TODO: Figure out what the `identity-button` class does -->
           <vbox class="identity-popup-security-connection identity-button">
             <hbox flex="1">
               <description class="identity-popup-connection-not-secure"
@@ -83,16 +85,150 @@
           </vbox>
 
         </vbox>
-        <button id="identity-popup-security-expander"
-                class="identity-popup-expander"
-                when-connection="not-secure secure secure-ev secure-cert-user-overridden cert-error-page https-only-error-page"
-                oncommand="gIdentityHandler.showSecuritySubView();"/>
+      </hbox>
+
+      <!-- Certificate Section -->
+      <hbox
+        id="identity-popup-certificate"
+        class="identity-popup-section"
+        hasexpander="true"
+        onclick="gIdentityHandler.showSecuritySubView();"
+      >
+        <vbox class="identity-popup-section-content" flex="1">
+          <image
+            class="identity-popup-section-icon"
+            style="--icon: url(chrome://browser/skin/certificate.svg)"
+          >
+          </image>
+
+          <div class="identity-popup-section-text-container">
+            <span
+              class="identity-popup-section-title"
+              data-l10n-id="identity-certificate-label"
+            ></span>
+            <span
+              id="identity-popup-section-subtitle"
+              class="identity-popup-section-subtitle"
+              data-l10n-id="identity-certificate-issuer"
+            ></span>
+          </div>
+        </vbox>
+        <button
+          id="identity-popup-certificate-expander"
+          class="identity-popup-expander"
+        />
+      </hbox>
+
+      <!-- Cookies Section -->
+      <hbox
+        id="identity-popup-cookies"
+        class="identity-popup-section"
+        hasexpander="true"
+      >
+        <vbox class="identity-popup-section-content" flex="1">
+          <image
+            class="identity-popup-section-icon"
+            style="--icon: url(chrome://browser/skin/cookies.svg)"
+          >
+          </image>
+
+          <div class="identity-popup-section-text-container">
+            <span
+              class="identity-popup-section-title"
+              data-l10n-id="identity-cookie-label"
+            ></span>
+          </div>
+        </vbox>
+        <button
+          id="identity-popup-cookies-expander"
+          class="identity-popup-expander"
+          oncommand="console.log('todo')"
+
+        />
+      </hbox>
+
+      <!-- Site Permissions Section -->
+      <hbox id="identity-popup-site-permissions-container">
+        <hbox
+          id="identity-popup-site-permissions"
+          class="identity-popup-section"
+          hasexpander="true"
+        >
+          <vbox class="identity-popup-section-content" flex="1">
+            <image
+              class="identity-popup-section-icon"
+              style="--icon: url(chrome://browser/skin/login.svg)"
+            >
+            </image>
+
+            <div class="identity-popup-section-text-container">
+              <span
+                class="identity-popup-section-title"
+                data-l10n-id="identity-site-permissions-label"
+              ></span>
+            </div>
+          </vbox>
+          <button
+            id="identity-popup-cookies-expander"
+            class="identity-popup-expander"
+            oncommand="console.log('todo')"
+          />
+        </hbox>
+
+        <hbox
+          class="identity-popup-section identity-popup-site-permissions-item"
+        >
+          <vbox class="identity-popup-section-content" flex="1">
+            <image
+              class="identity-popup-section-icon"
+              style="
+                --icon: url(chrome://browser/skin/notification-icons/location.svg);
+              "
+            >
+            </image>
+
+            <div class="identity-popup-section-text-container">
+              <span class="identity-popup-section-title">Location</span>
+              <span class="identity-popup-section-subtitle">Allowed</span>
+            </div>
+          </vbox>
+          <button
+            id="identity-popup-cookies-expander"
+            class="identity-popup-expander"
+            oncommand="console.log('todo')"
+          />
+        </hbox>
+
+        <hbox
+          class="identity-popup-section identity-popup-site-permissions-item"
+        >
+          <vbox class="identity-popup-section-content" flex="1">
+            <image
+              class="identity-popup-section-icon"
+              style="
+                --icon: url(chrome://browser/skin/notification-icons/javascript.svg);
+              "
+            >
+            </image>
+
+            <div class="identity-popup-section-text-container">
+              <span class="identity-popup-section-title">JavaScript</span>
+              <span class="identity-popup-section-subtitle">Allowed</span>
+            </div>
+          </vbox>
+          <button
+            id="identity-popup-cookies-expander"
+            class="identity-popup-expander"
+            oncommand="console.log('todo')"
+          />
+        </hbox>
       </hbox>
 
       <!-- Clear Site Data Button -->
       <vbox hidden="true"
             id="identity-popup-clear-sitedata-footer"
-            class="panel-footer panel-footer-menulike">
+            class="panel-footer panel-footer-menulike"
+            style="display: none">
         <button id="identity-popup-clear-sitedata-button"
                 data-l10n-id="identity-clear-site-data"
                 oncommand="gIdentityHandler.clearSiteData(event);"/>
@@ -180,7 +316,7 @@
                 oncommand="gIdentityHandler.enableMixedContentProtection()"/>
       </vbox>
 
-      <vbox id="identity-popup-more-info-footer" class="panel-footer panel-footer-menulike">
+      <vbox id="identity-popup-more-info-footer" class="panel-footer panel-footer-menulike" style="display: none">
         <!-- More Security Information -->
         <button id="identity-popup-more-info"
                 data-l10n-id="identity-more-info-link-text"
