diff --git a/browser/components/BrowserGlue.jsm b/browser/components/BrowserGlue.jsm
index 3ef1e1453591039f9c34edfe8b14af5d044d2266..fdc0e6e1b69306880635ab98168c0c348b1b04db 100644
--- a/browser/components/BrowserGlue.jsm
+++ b/browser/components/BrowserGlue.jsm
@@ -166,7 +166,7 @@ let JSPROCESSACTORS = {
               "RefreshBlocker",
               "all"
             );
-          } catch (ex) {}
+          } catch (ex) { }
         }
       });
     },
@@ -179,6 +179,20 @@ let JSPROCESSACTORS = {
  * available at https://firefox-source-docs.mozilla.org/dom/ipc/jsactors.html
  */
 let JSWINDOWACTORS = {
+  AboutHistory: {
+    parent: {
+      moduleURI: "resource:///actors/AboutHistoryParent.jsm",
+    },
+    child: {
+      moduleURI: "resource:///actors/AboutHistoryChild.jsm",
+
+      events: {
+        DOMWindowCreated: { capture: true },
+      },
+    },
+
+    matches: ["about:history"],
+  },
   AboutLogins: {
     parent: {
       moduleURI: "resource:///actors/AboutLoginsParent.jsm",
@@ -520,14 +534,19 @@ let JSWINDOWACTORS = {
       moduleURI: "resource:///actors/LightweightThemeChild.jsm",
       events: {
         pageshow: { mozSystemGroup: true },
+        DOMWindowCreated: {},
+        click: {},
       },
     },
     includeChrome: true,
     allFrames: true,
+    remoteTypes: ["privilegedabout"],
     matches: [
-      "about:home",
-      "about:newtab",
-      "about:welcome",
+      "about:*",
+      "about:certerror?*",
+      "about:neterror?*",
+      "chrome://browser/content/abouthistory/aboutHistory.xhtml",
+      "chrome://browser/content/preferences/new/newpreferences.html",
       "chrome://browser/content/syncedtabs/sidebar.xhtml",
       "chrome://browser/content/places/historySidebar.xhtml",
       "chrome://browser/content/places/bookmarksSidebar.xhtml",
@@ -765,8 +784,8 @@ let JSWINDOWACTORS = {
     !Services.startup.showedPreXULSkeletonUI &&
     Services.prefs.getCharPref(
       "extensions.activeThemeID",
-      "default-theme@mozilla.org"
-    ) != "default-theme@mozilla.org"
+      "dynamic@themes.dothq.co"
+    ) != "dynamic@themes.dothq.co"
   ) {
     return;
   }
@@ -1358,62 +1377,63 @@ BrowserGlue.prototype = {
     SessionStore.init();
 
     AddonManager.maybeInstallBuiltinAddon(
-      "firefox-compact-light@mozilla.org",
+      "light@themes.dothq.co",
       "1.1",
       "resource://builtin-themes/light/"
     );
     AddonManager.maybeInstallBuiltinAddon(
-      "firefox-compact-dark@mozilla.org",
+      "dark@themes.dothq.co",
       "1.1",
       "resource://builtin-themes/dark/"
     );
 
-    if (
-      AppConstants.NIGHTLY_BUILD &&
-      Services.prefs.getBoolPref("browser.proton.enabled", false)
-    ) {
-      // Temporarily install a fork of the light & dark themes to do development on for
-      // Proton. We only make this available if `browser.proton.enabled` is set
-      // to true, and we make sure to uninstall it again during shutdown.
-      const kProtonDarkThemeID = "firefox-compact-proton-dark@mozilla.org";
-      AddonManager.maybeInstallBuiltinAddon(
-        kProtonDarkThemeID,
-        "1.0",
-        "resource://builtin-themes/proton-dark/"
-      );
-
-      const kProtonLightThemeID = "firefox-compact-proton-light@mozilla.org";
-      AddonManager.maybeInstallBuiltinAddon(
-        kProtonLightThemeID,
-        "1.0",
-        "resource://builtin-themes/proton-light/"
-      );
-      AsyncShutdown.profileChangeTeardown.addBlocker(
-        "Uninstall Proton themes",
-        async () => {
-          for (let themeID of [kProtonDarkThemeID, kProtonLightThemeID]) {
-            try {
-              let addon = await AddonManager.getAddonByID(themeID);
-              await addon.uninstall();
-            } catch (e) {
-              Cu.reportError(`Failed to uninstall ${themeID} on shutdown`);
-            }
-          }
-        }
-      );
-    }
+//    if (
+//      AppConstants.NIGHTLY_BUILD &&
+//      Services.prefs.getBoolPref("browser.proton.enabled", false)
+//    ) {
+//      // Temporarily install a fork of the light & dark themes to do development on for
+//      // Proton. We only make this available if `browser.proton.enabled` is set
+//      // to true, and we make sure to uninstall it again during shutdown.
+//      const kProtonDarkThemeID = "firefox-compact-proton-dark@mozilla.org";
+//      AddonManager.maybeInstallBuiltinAddon(
+//        kProtonDarkThemeID,
+//        "1.0",
+//        "resource://builtin-themes/proton-dark/"
+//      );
+//
+//      const kProtonLightThemeID = "firefox-compact-proton-light@mozilla.org";
+//      AddonManager.maybeInstallBuiltinAddon(
+//        kProtonLightThemeID,
+//        "1.0",
+//        "resource://builtin-themes/proton-light/"
+//      );
+//      AsyncShutdown.profileChangeTeardown.addBlocker(
+//        "Uninstall Proton themes",
+//        async () => {
+//          for (let themeID of [kProtonDarkThemeID, kProtonLightThemeID]) {
+//            try {
+//              let addon = await AddonManager.getAddonByID(themeID);
+//              await addon.uninstall();
+//            } catch (e) {
+//              Cu.reportError(`Failed to uninstall ${themeID} on shutdown`);
+//            }
+//          }
+//        }
+//      );
+//    }
 
     AddonManager.maybeInstallBuiltinAddon(
-      "firefox-alpenglow@mozilla.org",
-      "1.2",
-      "resource://builtin-themes/alpenglow/"
+      "fusion@themes.dothq.co",
+      "0.1",
+      "resource://builtin-themes/fusion/"
     );
 
     if (AppConstants.MOZ_NORMANDY) {
       Normandy.init();
     }
 
-    SaveToPocket.init();
+    // This line enables pocket
+    // SaveToPocket.init();
 
     AboutHomeStartupCache.init();
 
@@ -2142,6 +2162,22 @@ BrowserGlue.prototype = {
     Services.wm.addListener(windowListener);
   },
 
+  _monitorQrCodesPref() {
+    const PREF = "browser.qrcodes.enabled";
+    const ID = "qr@dothq.co";
+    const _checkQrCodesPref = async () => {
+      let addon = await AddonManager.getAddonByID(ID);
+      let enabled = Services.prefs.getBoolPref(PREF, true);
+      if (enabled && !addon.isActive) {
+        await addon.enable({ allowSystemAddons: true });
+      } else if (!enabled && addon.isActive) {
+        await addon.disable({ allowSystemAddons: true });
+      }
+    };
+    Services.prefs.addObserver(PREF, _checkQrCodesPref);
+    _checkQrCodesPref();
+  },
+
   _showNewInstallModal() {
     // Allow other observers of the same topic to run while we open the dialog.
     Services.tm.dispatchToMainThread(() => {
@@ -2234,6 +2270,7 @@ BrowserGlue.prototype = {
     this._monitorHTTPSOnlyPref();
     this._monitorIonPref();
     this._monitorIonStudies();
+    this._monitorQrCodesPref();
 
     let pService = Cc["@mozilla.org/toolkit/profile-service;1"].getService(
       Ci.nsIToolkitProfileService
@@ -2425,7 +2462,7 @@ BrowserGlue.prototype = {
           let { setTimeout } = ChromeUtils.import(
             "resource://gre/modules/Timer.jsm"
           );
-          setTimeout(function() {
+          setTimeout(function () {
             Services.tm.idleDispatchToMainThread(
               Services.startup.trackStartupCrashEnd
             );
@@ -2633,7 +2670,7 @@ BrowserGlue.prototype = {
         this._gmpInstallManager = new obj.GMPInstallManager();
         // We don't really care about the results, if someone is interested they
         // can check the log.
-        this._gmpInstallManager.simpleCheckAndInstall().catch(() => {});
+        this._gmpInstallManager.simpleCheckAndInstall().catch(() => { });
       },
 
       () => {
@@ -2698,7 +2735,7 @@ BrowserGlue.prototype = {
 
   _addBreachAlertsPrefObserver() {
     const BREACH_ALERTS_PREF = "signon.management.page.breach-alerts.enabled";
-    const clearVulnerablePasswordsIfBreachAlertsDisabled = async function() {
+    const clearVulnerablePasswordsIfBreachAlertsDisabled = async function () {
       if (!Services.prefs.getBoolPref(BREACH_ALERTS_PREF)) {
         await LoginBreaches.clearAllPotentiallyVulnerablePasswords();
       }
@@ -2897,7 +2934,7 @@ BrowserGlue.prototype = {
       if (importBookmarksHTML) {
         importBookmarks = true;
       }
-    } catch (ex) {}
+    } catch (ex) { }
 
     // Support legacy bookmarks.html format for apps that depend on that format.
     let autoExportHTML = Services.prefs.getBoolPref(
@@ -2926,7 +2963,7 @@ BrowserGlue.prototype = {
           await this._backupBookmarks();
           importBookmarks = true;
         }
-      } catch (ex) {}
+      } catch (ex) { }
 
       // This may be reused later, check for "=== undefined" to see if it has
       // been populated already.
@@ -3085,14 +3122,14 @@ BrowserGlue.prototype = {
    * If a backup for today doesn't exist, this creates one.
    */
   _backupBookmarks: function BG__backupBookmarks() {
-    return (async function() {
+    return (async function () {
       let lastBackupFile = await PlacesBackups.getMostRecentBackup();
       // Should backup bookmarks if there are no backups or the maximum
       // interval between backups elapsed.
       if (
         !lastBackupFile ||
         new Date() - PlacesBackups.getDateForFile(lastBackupFile) >
-          BOOKMARKS_BACKUP_MIN_INTERVAL_DAYS * 86400000
+        BOOKMARKS_BACKUP_MIN_INTERVAL_DAYS * 86400000
       ) {
         let maxBackups = Services.prefs.getIntPref(
           "browser.bookmarks.max_backups"
@@ -3197,7 +3234,7 @@ BrowserGlue.prototype = {
     if (
       currentUIVersion < 65 &&
       Services.prefs.getCharPref("general.config.filename", "") ==
-        "dsengine.cfg"
+      "dsengine.cfg"
     ) {
       let searchInitializedPromise = new Promise(resolve => {
         if (Services.search.isInitialized) {
@@ -3413,7 +3450,7 @@ BrowserGlue.prototype = {
         !Services.prefs.getBoolPref("media.autoplay.allow-muted") &&
         !Services.prefs.prefHasUserValue("media.autoplay.default") &&
         Services.prefs.getIntPref("media.autoplay.default") ==
-          Ci.nsIAutoplay.BLOCKED
+        Ci.nsIAutoplay.BLOCKED
       ) {
         Services.prefs.setIntPref(
           "media.autoplay.default",
@@ -3559,8 +3596,8 @@ BrowserGlue.prototype = {
       } catch (error) {
         Cu.reportError(
           "Could not access the AddonManager to upgrade the profile. This is most " +
-            "likely because the upgrader is being run from an xpcshell test where " +
-            "the AddonManager is not initialized."
+          "likely because the upgrader is being run from an xpcshell test where " +
+          "the AddonManager is not initialized."
         );
       }
       Promise.resolve(addonPromise).then(addon => {
@@ -4388,7 +4425,7 @@ const ContentPermissionIntegration = {
   },
 };
 
-function ContentPermissionPrompt() {}
+function ContentPermissionPrompt() { }
 
 ContentPermissionPrompt.prototype = {
   classID: Components.ID("{d8903bf6-68d5-4e97-bcd1-e4d3012f721a}"),
@@ -4696,7 +4733,7 @@ var JawsScreenReaderVersionCheck = {
     if (!win || !win.gBrowser || !win.gBrowser.selectedBrowser) {
       Services.console.logStringMessage(
         "Content access support for older versions of JAWS is disabled " +
-          "due to compatibility issues with this version of Firefox."
+        "due to compatibility issues with this version of Firefox."
       );
       this._prompted = false;
       return;
@@ -5292,7 +5329,7 @@ var AboutHomeStartupCache = {
 
         this.log.trace(
           "Writing the page data is complete. Now opening the " +
-            "script output stream."
+          "script output stream."
         );
 
         let scriptOutputStream;
@@ -5437,7 +5474,7 @@ var AboutHomeStartupCache = {
       if (this._cacheDeferred) {
         this.log.error(
           "A privileged about content process shut down while cache streams " +
-            "were still en route."
+          "were still en route."
         );
         // The crash occurred while we were waiting on cache input streams to
         // be returned to us. Resolve with null streams instead.
@@ -5535,7 +5572,7 @@ var AboutHomeStartupCache = {
       ) {
         this.log.error(
           "Somehow got a success result despite having never " +
-            "successfully sent down the cache streams"
+          "successfully sent down the cache streams"
         );
         this.recordResult(this._cacheDeferredResultScalar);
       } else {
