diff --git a/browser/base/content/browser.xhtml b/browser/base/content/browser.xhtml
index 0a1090292f54612dbf0f564c772f12ae0056dc67..d1eee5f91cbba57c5f790910f45d39fc08162116 100644
--- a/browser/base/content/browser.xhtml
+++ b/browser/base/content/browser.xhtml
@@ -80,6 +80,10 @@
   <link rel="localization" href="toolkit/printing/printUI.ftl"/>
   <link rel="localization" href="browser/tabbrowser.ftl"/>
 
+  <!-- Custom localisation for Dot Browser -->
+  <link rel="localization" href="browser/qr-codes.ftl"/>
+  <link rel="localization" href="browser/toolbarContextMenu.ftl"/>
+
   <title data-l10n-id="browser-main-window-title"></title>
 
 # All JS files which are needed by browser.xhtml and other top level windows to
@@ -134,17 +138,20 @@
       <menuitem id="context_openANewTab" data-lazy-l10n-id="tab-context-new-tab"
                 oncommand="gBrowser.addAdjacentNewTab(TabContextMenu.contextTab)"/>
       <menuseparator/>
-      <menuitem id="context_reloadTab" data-lazy-l10n-id="reload-tab"
-                oncommand="gBrowser.reloadTab(TabContextMenu.contextTab);"/>
+      <menuitem id="context_reloadTab" class="menuitem-iconic" data-lazy-l10n-id="reload-tab"
+                oncommand="gBrowser.reloadTab(TabContextMenu.contextTab);"
+                image="chrome://browser/skin/reload.svg"/>
       <menuitem id="context_reloadSelectedTabs" data-lazy-l10n-id="reload-tabs" hidden="true"
                 oncommand="gBrowser.reloadMultiSelectedTabs();"/>
-      <menuitem id="context_toggleMuteTab" oncommand="TabContextMenu.contextTab.toggleMuteAudio();"/>
+      <menuitem id="context_toggleMuteTab" class="menuitem-iconic" oncommand="TabContextMenu.contextTab.toggleMuteAudio();" image="chrome://global/skin/pictureinpicture/audio-mute.svg"/>
       <menuitem id="context_toggleMuteSelectedTabs" hidden="true"
                 oncommand="gBrowser.toggleMuteAudioOnMultiSelectedTabs(TabContextMenu.contextTab);"/>
-      <menuitem id="context_pinTab" data-lazy-l10n-id="pin-tab"
-                oncommand="gBrowser.pinTab(TabContextMenu.contextTab);"/>
-      <menuitem id="context_unpinTab" data-lazy-l10n-id="unpin-tab" hidden="true"
-                oncommand="gBrowser.unpinTab(TabContextMenu.contextTab);"/>
+      <menuitem id="context_pinTab" class="menuitem-iconic" data-lazy-l10n-id="pin-tab"
+                oncommand="gBrowser.pinTab(TabContextMenu.contextTab);"
+                image="chrome://browser/skin/pin-tab.svg"/>
+      <menuitem id="context_unpinTab" class="menuitem-iconic" data-lazy-l10n-id="unpin-tab" hidden="true"
+                oncommand="gBrowser.unpinTab(TabContextMenu.contextTab);"
+                image="chrome://browser/skin/unpin-tab.svg"/>
       <menuitem id="context_pinSelectedTabs" data-lazy-l10n-id="pin-selected-tabs" hidden="true"
                 oncommand="gBrowser.pinMultiSelectedTabs();"/>
       <menuitem id="context_unpinSelectedTabs" data-lazy-l10n-id="unpin-selected-tabs" hidden="true"
@@ -160,7 +167,9 @@
                 oncommand="PlacesUIUtils.showBookmarkPagesDialog(PlacesCommandHook.uniqueSelectedPages);"/>
       <menuitem id="context_bookmarkTab"
                 data-lazy-l10n-id="bookmark-tab"
-                oncommand="PlacesUIUtils.showBookmarkPagesDialog(PlacesCommandHook.getUniquePages([TabContextMenu.contextTab]));"/>
+                class="menuitem-iconic"
+                oncommand="PlacesUIUtils.showBookmarkPagesDialog(PlacesCommandHook.getUniquePages([TabContextMenu.contextTab]));"
+                image="chrome://browser/skin/bookmark.svg"/>
       <menu id="context_moveTabOptions"
             data-lazy-l10n-id="tab-context-move-tabs"
             data-l10n-args='{"tabCount": 1}'>
@@ -195,7 +204,9 @@
       <menuitem id="context_closeTab"
                 data-lazy-l10n-id="tab-context-close-tabs"
                 data-l10n-args='{"tabCount": 1}'
-                oncommand="TabContextMenu.closeContextTabs();"/>
+                oncommand="TabContextMenu.closeContextTabs();"
+                class="menuitem-iconic"
+                image="chrome://browser/skin/stop.svg"/>
       <menu id="context_closeTabOptions"
             data-lazy-l10n-id="tab-context-close-multiple-tabs">
         <menupopup id="closeTabOptions">
@@ -210,7 +221,9 @@
       <menuitem id="context_undoCloseTab"
                 data-lazy-l10n-id="tab-context-reopen-closed-tabs"
                 data-l10n-args='{"tabCount": 1}'
-                observes="History:UndoCloseTab"/>
+                observes="History:UndoCloseTab"
+                class="menuitem-iconic"
+                image="chrome://global/skin/icons/undo.svg"/>
     </menupopup>
 
     <!-- bug 415444/582485: event.stopPropagation is here for the cloned version
@@ -405,20 +418,6 @@
            flip="slide"
            orient="vertical"
            position="bottomcenter topleft">
-      <toolbarbutton id="sidebar-switcher-bookmarks"
-                     data-l10n-id="sidebar-menu-bookmarks"
-                     class="subviewbutton"
-                     key="viewBookmarksSidebarKb"
-                     oncommand="SidebarUI.show('viewBookmarksSidebar');"/>
-      <toolbarbutton id="sidebar-switcher-history"
-                     data-l10n-id="sidebar-menu-history"
-                     class="subviewbutton"
-                     key="key_gotoHistory"
-                     oncommand="SidebarUI.show('viewHistorySidebar');"/>
-      <toolbarbutton id="sidebar-switcher-tabs"
-                     data-l10n-id="sidebar-menu-synced-tabs"
-                     class="subviewbutton sync-ui-item"
-                     oncommand="SidebarUI.show('viewTabsSidebar');"/>
       <toolbarseparator/>
       <!-- Extension toolbarbuttons go here. -->
       <toolbarseparator id="sidebar-extensions-separator"/>
@@ -498,8 +497,9 @@
                     correctly. -->
       <menuitem oncommand="gCustomizeMode.enter()"
                 observes="cmd_CustomizeToolbars"
-                class="viewCustomizeToolbar"
-                data-lazy-l10n-id="toolbar-context-menu-view-customize-toolbar-2"/>
+                class="viewCustomizeToolbar menuitem-iconic"
+                data-lazy-l10n-id="toolbar-context-menu-view-customize-toolbar-2"
+                image="chrome://browser/skin/customize.svg"/>
     </menupopup>
 
     <menupopup id="blockedPopupOptions"
@@ -555,6 +555,21 @@
 
 #include ../../components/places/content/placesContextMenu.inc.xhtml
 
+    <menupopup id="sidebar-context-menu">
+      <menuitem class="menuitem-iconic"
+                observes="cmd_CustomizeToolbars"
+                data-l10n-id="toolbar-context-menu-view-customize-toolbar"
+                image="chrome://browser/skin/customize.svg"
+      />
+    </menupopup>
+
+    <menupopup id="sidebar-item-context-menu">
+      <menuitem class="menuitem-iconic"
+                data-l10n-id="sidebar-context-menu-remove-item"
+                image="chrome://global/skin/icons/delete.svg"
+      />
+    </menupopup>
+
     <panel id="ctrlTab-panel" hidden="true" norestorefocus="true" level="top">
       <hbox id="ctrlTab-previews"/>
       <hbox id="ctrlTab-showAll-container" pack="center"/>
@@ -632,7 +640,6 @@
 
 #include ../../components/customizableui/content/panelUI.inc.xhtml
 #include ../../components/controlcenter/content/identityPanel.inc.xhtml
-#include ../../components/controlcenter/content/permissionPanel.inc.xhtml
 #include ../../components/controlcenter/content/protectionsPanel.inc.xhtml
 #include ../../components/downloads/content/downloadsPanel.inc.xhtml
 #include ../../../devtools/startup/enableDevToolsPopup.inc.xhtml
@@ -763,7 +770,7 @@
                         oncommand="gSync.toggleAccountPanel('PanelUI-fxa', this, event)"/>
         </toolbaritem>
         <toolbarseparator class="sync-ui-item"/>
-        <toolbaritem>
+        <toolbaritem hidden="true">
           <toolbarbutton id="appMenu-protection-report-button"
                class="subviewbutton subviewbutton-iconic"
                oncommand="gProtectionsHandler.openProtections(); gProtectionsHandler.recordClick('open_full_report', null, 'app_menu');">
@@ -770,7 +785,12 @@
             </label>
           </toolbarbutton>
         </toolbaritem>
-        <toolbarseparator id="appMenu-tp-separator"/>
+        <toolbarseparator id="appMenu-tp-separator" hidden="true"/>
+        <toolbarbutton id="appMenu-new-tab-button2"
+                       class="subviewbutton subviewbutton-iconic"
+                       label="&tabCmd.label;"
+                       key="key_newNavigatorTab"
+                       command="cmd_newNavigatorTab"/>
         <toolbarbutton id="appMenu-new-window-button"
                        class="subviewbutton subviewbutton-iconic"
                        label="&newNavigatorCmd.label;"
@@ -796,7 +816,10 @@
         <toolbarbutton id="appMenuRestoreLastSession"
                        label="&appMenuHistory.restoreSession.label;"
                        class="subviewbutton subviewbutton-iconic"
-                       command="Browser:RestoreLastSession"/>
+                       command="Browser:RestoreLastSession"
+                       hidden="true"/>
+          <toolbarseparator hidden="true"/>
+          <toolbaritem id="appMenu-zoom-controls" class="toolbaritem-combined-buttons" closemenu="none" hidden="true">
         <toolbarseparator/>
         <toolbaritem id="appMenu-zoom-controls" class="toolbaritem-combined-buttons" closemenu="none">
           <!-- Use a spacer, because panel sizing code gets confused when using CSS methods. -->
           <spacer class="before-label"/>
           <label value="&fullZoom.label;"/>
@@ -832,8 +845,8 @@
                          onclick="if (event.button == 0) this.closest('panel').hidePopup();"
                          tooltip="dynamic-shortcut-tooltip"/>
         </toolbaritem>
-        <toolbarseparator/>
-        <toolbaritem id="appMenu-edit-controls" class="toolbaritem-combined-buttons" closemenu="none">
+        <toolbarseparator hidden="true"/>
+        <toolbaritem id="appMenu-edit-controls" class="toolbaritem-combined-buttons" closemenu="none" hidden="true">
           <!-- Use a spacer, because panel sizing code gets confused when using CSS methods. -->
           <spacer class="before-label"/>
           <label value="&editMenu.label;"/>
@@ -853,18 +876,34 @@
                        class="subviewbutton subviewbutton-iconic subviewbutton-nav"
                        label="&places.library.title;"
                        closemenu="none"
-                       oncommand="PanelUI.showSubView('appMenu-libraryView', this)"/>
+                       oncommand="PanelUI.showSubView('appMenu-libraryView', this)"
+                       hidden="true"/>
         <toolbarbutton id="appMenu-logins-button"
                        class="subviewbutton subviewbutton-iconic"
                        label="&logins.label;"
                        oncommand="LoginHelper.openPasswordManager(window, { entryPoint: 'mainmenu' })"
-                       />
+                       hidden="true"/>
         <toolbarbutton id="appMenu-addons-button"
                        class="subviewbutton subviewbutton-iconic"
                        data-l10n-id="appmenuitem-addons-and-themes"
                        key="key_openAddons"
                        command="Tools:Addons"
-                       />
+                       hidden="true"/>
+        <toolbarbutton id="appMenu-library-history-button"
+                       class="subviewbutton subviewbutton-iconic subviewbutton-nav"
+                       label="&historyMenu.label;"
+                       closemenu="none"
+                       oncommand="PanelUI.showSubView('PanelUI-history', this)"/>
+        <toolbarbutton id="appMenu-library-bookmarks-button"
+                       class="subviewbutton subviewbutton-iconic subviewbutton-nav"
+                       data-l10n-id="library-bookmarks-menu"
+                       closemenu="none"
+                       oncommand="BookmarkingUI.showSubView(this);"/>
+        <toolbarbutton id="appMenu-library-downloads-button"
+                       class="subviewbutton subviewbutton-iconic subviewbutton-nav"
+                       label="&libraryDownloads.label;"
+                       closemenu="none"
+                       oncommand="DownloadsSubview.show(this);"/>
         <toolbarbutton id="appMenu-preferences-button"
                        class="subviewbutton subviewbutton-iconic"
 #ifdef XP_WIN
@@ -943,6 +982,7 @@
                        key="key_quitApplication"
                        command="cmd_quitApplication"/>
 #endif
+        </toolbaritem>
       </vbox>
     </panelview>
 
@@ -1006,17 +1046,6 @@
                        label="&libraryDownloads.label;"
                        key="key_openDownloads"
                        command="Tools:Downloads"/>
-        <toolbarbutton id="appMenu-passwords-button"
-                       class="subviewbutton"
-                       data-l10n-id="appmenuitem-passwords"
-                       oncommand="LoginHelper.openPasswordManager(window, { entryPoint: 'mainmenu' })"
-                       />
-        <toolbarbutton id="appMenu-extensions-themes-button"
-                       class="subviewbutton"
-                       data-l10n-id="appmenuitem-addons-and-themes"
-                       key="key_openAddons"
-                       command="Tools:Addons"
-                       />
+                       oncommand="openPreferences()"/>
         <toolbarbutton id="appMenu-customize-button"
                        class="subviewbutton subviewbutton-iconic"
                        data-l10n-id="appmenu-customizetoolbar"
@@ -910,11 +938,13 @@
 #endif
                        />
         <toolbarseparator/>
         <toolbarbutton id="appMenu-print-button2"
                        class="subviewbutton"
@@ -1038,7 +1067,7 @@
                        data-l10n-id="appmenuitem-find-in-page"
                        key="key_find"
                        command="cmd_find"/>
+        <toolbarseparator/>
         <toolbarbutton id="appMenu-more-button"
                        class="subviewbutton subviewbutton-nav"
                        label="&moreMenu.label;"
@@ -1646,27 +1676,28 @@
   <box>
   <toolbox id="navigator-toolbox" flex="1">
 
-    <vbox id="titlebar">
-      <!-- Menu -->
-      <toolbar type="menubar" id="toolbar-menubar"
-               class="browser-toolbar chromeclass-menubar titlebar-color"
-               customizable="true"
-               mode="icons"
+    <!-- Menu -->
+    <toolbar type="menubar" id="toolbar-menubar"
+              class="browser-toolbar chromeclass-menubar titlebar-color"
+              customizable="true"
+              mode="icons"
 #ifdef MENUBAR_CAN_AUTOHIDE
-               toolbarname="&menubarCmd.label;"
-               accesskey="&menubarCmd.accesskey;"
-               autohide="true"
+              toolbarname="&menubarCmd.label;"
+              accesskey="&menubarCmd.accesskey;"
+              autohide="true"
 #endif
-               context="toolbar-context-menu">
-        <toolbaritem id="menubar-items" align="center">
+              context="toolbar-context-menu">
+      <toolbaritem id="menubar-items" align="center">
+        <image class="menubar-product-icon"></image>
 # The entire main menubar is placed into browser-menubar.inc, so that it can be
 # shared with other top level windows in macWindow.inc.xhtml.
 #include browser-menubar.inc
-        </toolbaritem>
-        <spacer flex="1" skipintoolbarset="true" style="-moz-box-ordinal-group: 1000;"/>
+      </toolbaritem>
+      <spacer flex="1" skipintoolbarset="true" style="-moz-box-ordinal-group: 1000;"/>
 #include titlebar-items.inc.xhtml
-      </toolbar>
+    </toolbar>
 
+    <vbox id="titlebar">
       <toolbar id="TabsToolbar"
                class="browser-toolbar titlebar-color"
                fullscreentoolbar="true"
@@ -2349,6 +2380,44 @@
   </box>
 
   <hbox flex="1" id="browser">
+    <vbox id="sidebar-iconbar" hidden="true" context="sidebar-context-menu">
+      <vbox class="sidebar-iconbar-container">
+        <toolbarbutton id="sidebar-switcher-bookmarks"
+              class="sidebar-iconbar-item"
+              oncommand="SidebarUI.toggle('viewBookmarksSidebar');"
+              context="sidebar-item-context-menu"
+              data-l10n-id="sidebar-menu-bookmarks"
+        />
+        <toolbarbutton id="sidebar-switcher-history"
+              class="sidebar-iconbar-item"
+              oncommand="SidebarUI.toggle('viewHistorySidebar');"
+              context="sidebar-item-context-menu"
+              data-l10n-id="sidebar-menu-history"
+        />
+        <toolbarbutton id="sidebar-switcher-tabs"
+              class="sidebar-iconbar-item"
+              oncommand="SidebarUI.toggle('viewTabsSidebar');"
+              context="sidebar-item-context-menu"
+              data-l10n-id="sidebar-menu-synced-tabs"
+        />
+      </vbox>
+
+      <vbox class="sidebar-iconbar-container">
+        <toolbarbutton id="sidebar-switcher-accounts"
+              class="sidebar-iconbar-item"
+              oncommand="SidebarUI.toggle('viewAccountsSidebar');"
+              context="sidebar-item-context-menu"
+              data-l10n-id="sidebar-menu-accounts"
+        />
+        <toolbarbutton id="sidebar-switcher-settings"
+              class="sidebar-iconbar-item"
+              oncommand="openPreferences();"
+              context="sidebar-item-context-menu"
+              data-l10n-id="sidebar-menu-settings"
+        />
+      </vbox>
+    </vbox>
+
     <vbox id="sidebar-box" hidden="true" class="chromeclass-extrachrome">
       <box id="sidebar-header" align="center">
         <toolbarbutton id="sidebar-switcher-target" flex="1" class="tabbable">
@@ -2368,7 +2435,7 @@
         <toolbarbutton id="sidebar-close" class="close-icon tabbable" tooltiptext="&sidebarCloseButton.tooltip;" oncommand="SidebarUI.hide();"/>
       </box>
       <browser id="sidebar" flex="1" autoscroll="false" disablehistory="true" disablefullscreen="true"
-                style="min-width: 14em; width: 18em; max-width: 36em;" tooltip="aHTMLTooltip"/>
+                style="min-width: 14em; width: 24em; max-width: 36em;" tooltip="aHTMLTooltip"/>
     </vbox>
 
     <splitter id="sidebar-splitter" class="chromeclass-extrachrome sidebar-splitter" hidden="true"/>
