diff --git a/browser/components/urlbar/UrlbarInput.jsm b/browser/components/urlbar/UrlbarInput.jsm
index bd0255e37d399d03bb541e8e180615753c6441f9..8aa3b7f1f73a1fe67db9ca607a1492afa7fa8d6e 100644
--- a/browser/components/urlbar/UrlbarInput.jsm
+++ b/browser/components/urlbar/UrlbarInput.jsm
@@ -2502,6 +2502,8 @@ class UrlbarInput {
       .GetStringFromName("pasteAndGo.label");
     pasteAndGo.setAttribute("label", label);
     pasteAndGo.setAttribute("anonid", "paste-and-go");
+    pasteAndGo.setAttribute("image", "chrome://global/skin/icons/search-glass.svg");
+    pasteAndGo.classList.add("menuitem-iconic");
     pasteAndGo.addEventListener("command", () => {
       this._suppressStartQuery = true;
 
