diff --git a/browser/base/content/browser.xhtml b/browser/base/content/browser.xhtml
index 57a95856fa4beb0fa68dc9e2a88bb29daa2b4769..76ca83a39caf1037e137a60f620f2592f6248955 100644
--- a/browser/base/content/browser.xhtml
+++ b/browser/base/content/browser.xhtml
@@ -130,17 +130,20 @@
     <menupopup id="tabContextMenu"
                onpopupshowing="if (event.target == this) TabContextMenu.updateContextMenu(this);"
                onpopuphidden="if (event.target == this) TabContextMenu.contextTab = null;">
-      <menuitem id="context_reloadTab" data-lazy-l10n-id="reload-tab"
-                oncommand="gBrowser.reloadTab(TabContextMenu.contextTab);"/>
+      <menuitem id="context_reloadTab" class="menuitem-iconic" data-lazy-l10n-id="reload-tab"
+                oncommand="gBrowser.reloadTab(TabContextMenu.contextTab);"
+                image="chrome://browser/skin/reload.svg"/>
       <menuitem id="context_reloadSelectedTabs" data-lazy-l10n-id="reload-tabs" hidden="true"
                 oncommand="gBrowser.reloadMultiSelectedTabs();"/>
-      <menuitem id="context_toggleMuteTab" oncommand="TabContextMenu.contextTab.toggleMuteAudio();"/>
+      <menuitem id="context_toggleMuteTab" class="menuitem-iconic" oncommand="TabContextMenu.contextTab.toggleMuteAudio();" image="chrome://global/skin/pictureinpicture/audio-mute.svg"/>
       <menuitem id="context_toggleMuteSelectedTabs" hidden="true"
                 oncommand="gBrowser.toggleMuteAudioOnMultiSelectedTabs(TabContextMenu.contextTab);"/>
-      <menuitem id="context_pinTab" data-lazy-l10n-id="pin-tab"
-                oncommand="gBrowser.pinTab(TabContextMenu.contextTab);"/>
-      <menuitem id="context_unpinTab" data-lazy-l10n-id="unpin-tab" hidden="true"
-                oncommand="gBrowser.unpinTab(TabContextMenu.contextTab);"/>
+      <menuitem id="context_pinTab" class="menuitem-iconic" data-lazy-l10n-id="pin-tab"
+                oncommand="gBrowser.pinTab(TabContextMenu.contextTab);"
+                image="chrome://browser/skin/pin-tab.svg"/>
+      <menuitem id="context_unpinTab" class="menuitem-iconic" data-lazy-l10n-id="unpin-tab" hidden="true"
+                oncommand="gBrowser.unpinTab(TabContextMenu.contextTab);"
+                image="chrome://browser/skin/unpin-tab.svg"/>
       <menuitem id="context_pinSelectedTabs" data-lazy-l10n-id="pin-selected-tabs" hidden="true"
                 oncommand="gBrowser.pinMultiSelectedTabs();"/>
       <menuitem id="context_unpinSelectedTabs" data-lazy-l10n-id="unpin-selected-tabs" hidden="true"
@@ -156,7 +159,9 @@
                 oncommand="PlacesUIUtils.showBookmarkPagesDialog(PlacesCommandHook.uniqueSelectedPages);"/>
       <menuitem id="context_bookmarkTab"
                 data-lazy-l10n-id="bookmark-tab"
-                oncommand="PlacesUIUtils.showBookmarkPagesDialog(PlacesCommandHook.getUniquePages([TabContextMenu.contextTab]));"/>
+                class="menuitem-iconic"
+                oncommand="PlacesUIUtils.showBookmarkPagesDialog(PlacesCommandHook.getUniquePages([TabContextMenu.contextTab]));"
+                image="chrome://browser/skin/bookmark.svg"/>
       <menu id="context_moveTabOptions"
             data-lazy-l10n-id="tab-context-move-tabs"
             data-l10n-args='{"tabCount": 1}'>
@@ -191,7 +196,9 @@
       <menuitem id="context_closeTab"
                 data-lazy-l10n-id="tab-context-close-tabs"
                 data-l10n-args='{"tabCount": 1}'
-                oncommand="TabContextMenu.closeContextTabs();"/>
+                oncommand="TabContextMenu.closeContextTabs();"
+                class="menuitem-iconic"
+                image="chrome://browser/skin/stop.svg"/>
       <menu id="context_closeTabOptions"
             data-lazy-l10n-id="tab-context-close-multiple-tabs">
         <menupopup id="closeTabOptions">
@@ -206,7 +213,9 @@
       <menuitem id="context_undoCloseTab"
                 data-lazy-l10n-id="tab-context-reopen-closed-tabs"
                 data-l10n-args='{"tabCount": 1}'
-                observes="History:UndoCloseTab"/>
+                observes="History:UndoCloseTab"
+                class="menuitem-iconic"
+                image="chrome://global/skin/icons/undo.svg"/>
     </menupopup>
 
     <!-- bug 415444/582485: event.stopPropagation is here for the cloned version
@@ -490,8 +499,9 @@
                     correctly. -->
       <menuitem oncommand="gCustomizeMode.enter()"
                 observes="cmd_CustomizeToolbars"
-                class="viewCustomizeToolbar"
-                data-lazy-l10n-id="toolbar-context-menu-view-customize-toolbar-2"/>
+                data-lazy-l10n-id="toolbar-context-menu-view-customize-toolbar-2"
+                class="viewCustomizeToolbar menuitem-iconic"
+                image="chrome://browser/skin/customize.svg"/>
     </menupopup>
 
     <menupopup id="blockedPopupOptions"
@@ -770,7 +780,7 @@
                         oncommand="gSync.toggleAccountPanel(this, event)"/>
         </toolbaritem>
         <toolbarseparator class="sync-ui-item"/>
-        <toolbaritem>
+        <toolbaritem hidden="true">
           <toolbarbutton id="appMenu-protection-report-button"
                class="subviewbutton subviewbutton-iconic"
                oncommand="gProtectionsHandler.openProtections(); gProtectionsHandler.recordClick('open_full_report', null, 'app_menu');">
@@ -781,7 +791,12 @@
             </label>
           </toolbarbutton>
         </toolbaritem>
-        <toolbarseparator id="appMenu-tp-separator"/>
+        <toolbarseparator id="appMenu-tp-separator" hidden="true"/>
+        <toolbarbutton id="appMenu-new-tab-button2"
+                       class="subviewbutton subviewbutton-iconic"
+                       label="&tabCmd.label;"
+                       key="key_newNavigatorTab"
+                       command="cmd_newNavigatorTab"/>
         <toolbarbutton id="appMenu-new-window-button"
                        class="subviewbutton subviewbutton-iconic"
                        label="&newNavigatorCmd.label;"
@@ -807,9 +822,10 @@
         <toolbarbutton id="appMenuRestoreLastSession"
                        label="&appMenuHistory.restoreSession.label;"
                        class="subviewbutton subviewbutton-iconic"
-                       command="Browser:RestoreLastSession"/>
-        <toolbarseparator/>
-        <toolbaritem id="appMenu-zoom-controls" class="toolbaritem-combined-buttons" closemenu="none">
+                       command="Browser:RestoreLastSession"
+                       hidden="true"/>
+          <toolbarseparator hidden="true"/>
+          <toolbaritem id="appMenu-zoom-controls" class="toolbaritem-combined-buttons" closemenu="none" hidden="true">
           <!-- Use a spacer, because panel sizing code gets confused when using CSS methods. -->
           <spacer class="before-label"/>
           <label value="&fullZoom.label;"/>
@@ -839,8 +855,8 @@
                          onclick="if (event.button == 0) this.closest('panel').hidePopup();"
                          tooltip="dynamic-shortcut-tooltip"/>
         </toolbaritem>
-        <toolbarseparator/>
-        <toolbaritem id="appMenu-edit-controls" class="toolbaritem-combined-buttons" closemenu="none">
+        <toolbarseparator hidden="true"/>
+        <toolbaritem id="appMenu-edit-controls" class="toolbaritem-combined-buttons" closemenu="none" hidden="true">
           <!-- Use a spacer, because panel sizing code gets confused when using CSS methods. -->
           <spacer class="before-label"/>
           <label value="&editMenu.label;"/>
@@ -864,18 +880,34 @@
                        class="subviewbutton subviewbutton-iconic subviewbutton-nav"
                        label="&places.library.title;"
                        closemenu="none"
-                       oncommand="PanelUI.showSubView('appMenu-libraryView', this)"/>
+                       oncommand="PanelUI.showSubView('appMenu-libraryView', this)"
+                       hidden="true"/>
         <toolbarbutton id="appMenu-logins-button"
                        class="subviewbutton subviewbutton-iconic"
                        label="&logins.label;"
                        oncommand="LoginHelper.openPasswordManager(window, { entryPoint: 'mainmenu' })"
-                       />
+                       hidden="true"/>
         <toolbarbutton id="appMenu-addons-button"
                        class="subviewbutton subviewbutton-iconic"
                        data-l10n-id="appmenuitem-addons-and-themes"
                        key="key_openAddons"
                        command="Tools:Addons"
-                       />
+                       hidden="true"/>
+        <toolbarbutton id="appMenu-library-history-button"
+                       class="subviewbutton subviewbutton-iconic subviewbutton-nav"
+                       label="&historyMenu.label;"
+                       closemenu="none"
+                       oncommand="PanelUI.showSubView('PanelUI-history', this)"/>
+        <toolbarbutton id="appMenu-library-bookmarks-button"
+                       class="subviewbutton subviewbutton-iconic subviewbutton-nav"
+                       data-l10n-id="library-bookmarks-menu"
+                       closemenu="none"
+                       oncommand="BookmarkingUI.showSubView(this);"/>
+        <toolbarbutton id="appMenu-library-downloads-button"
+                       class="subviewbutton subviewbutton-iconic subviewbutton-nav"
+                       label="&libraryDownloads.label;"
+                       closemenu="none"
+                       oncommand="DownloadsSubview.show(this);"/>
         <toolbarbutton id="appMenu-preferences-button"
                        class="subviewbutton subviewbutton-iconic"
 #ifdef XP_WIN
@@ -922,6 +954,7 @@
                        label="&findOnCmd.label;"
                        key="key_find"
                        command="cmd_find"/>
+        <toolbarseparator/>
         <toolbarbutton id="appMenu-more-button"
                        class="subviewbutton subviewbutton-nav"
                        label="&moreMenu.label;"
