diff --git a/browser/components/newtab/aboutwelcome/lib/AboutWelcomeDefaults.jsm b/browser/components/newtab/aboutwelcome/lib/AboutWelcomeDefaults.jsm
index 70bcd50defb3951bb2528b2b4ea149ab8e19efb6..6547ba1f3a2fe14edc308ef79b599d320b0fc4fe 100644
--- a/browser/components/newtab/aboutwelcome/lib/AboutWelcomeDefaults.jsm
+++ b/browser/components/newtab/aboutwelcome/lib/AboutWelcomeDefaults.jsm
@@ -159,18 +159,6 @@ const DEFAULT_WELCOME_CONTENT = {
                 string_id: "onboarding-multistage-theme-description-dark",
               },
             },
-            {
-              theme: "alpenglow",
-              label: {
-                string_id: "onboarding-multistage-theme-label-alpenglow",
-              },
-              tooltip: {
-                string_id: "onboarding-multistage-theme-tooltip-alpenglow-2",
-              },
-              description: {
-                string_id: "onboarding-multistage-theme-description-alpenglow",
-              },
-            },
           ],
         },
         primary_button: {
