diff --git a/browser/base/content/browser.xhtml b/browser/base/content/browser.xhtml
index 0a1090292f54612dbf0f564c772f12ae0056dc67..39b40215218e471290ba76ab8b984f7a5ce3d8ba 100644
--- a/browser/base/content/browser.xhtml
+++ b/browser/base/content/browser.xhtml
@@ -763,7 +763,7 @@
                         oncommand="gSync.toggleAccountPanel('PanelUI-fxa', this, event)"/>
         </toolbaritem>
         <toolbarseparator class="sync-ui-item"/>
-        <toolbaritem>
+        <toolbaritem hidden="true">
           <toolbarbutton id="appMenu-protection-report-button"
                class="subviewbutton subviewbutton-iconic"
                oncommand="gProtectionsHandler.openProtections(); gProtectionsHandler.recordClick('open_full_report', null, 'app_menu');">
@@ -774,7 +774,12 @@
             </label>
           </toolbarbutton>
         </toolbaritem>
-        <toolbarseparator id="appMenu-tp-separator"/>
+        <toolbarseparator id="appMenu-tp-separator" hidden="true"/>
+        <toolbarbutton id="appMenu-new-tab-button"
+                       class="subviewbutton subviewbutton-iconic"
+                       label="&newTabCmd.label;"
+                       key="key_newNavigatorTab"
+                       oncommand="BrowserOpenTab()"/>
         <toolbarbutton id="appMenu-new-window-button"
                        class="subviewbutton subviewbutton-iconic"
                        label="&newNavigatorCmd.label;"
@@ -800,9 +805,10 @@
         <toolbarbutton id="appMenuRestoreLastSession"
                        label="&appMenuHistory.restoreSession.label;"
                        class="subviewbutton subviewbutton-iconic"
-                       command="Browser:RestoreLastSession"/>
-        <toolbarseparator/>
-        <toolbaritem id="appMenu-zoom-controls" class="toolbaritem-combined-buttons" closemenu="none">
+                       command="Browser:RestoreLastSession"
+                       hidden="true"/>
+          <toolbarseparator hidden="true"/>
+          <toolbaritem id="appMenu-zoom-controls" class="toolbaritem-combined-buttons" closemenu="none" hidden="true">
           <!-- Use a spacer, because panel sizing code gets confused when using CSS methods. -->
           <spacer class="before-label"/>
           <label value="&fullZoom.label;"/>
@@ -832,8 +838,8 @@
                          onclick="if (event.button == 0) this.closest('panel').hidePopup();"
                          tooltip="dynamic-shortcut-tooltip"/>
         </toolbaritem>
-        <toolbarseparator/>
-        <toolbaritem id="appMenu-edit-controls" class="toolbaritem-combined-buttons" closemenu="none">
+        <toolbarseparator hidden="true"/>
+        <toolbaritem id="appMenu-edit-controls" class="toolbaritem-combined-buttons" closemenu="none" hidden="true">
           <!-- Use a spacer, because panel sizing code gets confused when using CSS methods. -->
           <spacer class="before-label"/>
           <label value="&editMenu.label;"/>
@@ -857,18 +863,34 @@
                        class="subviewbutton subviewbutton-iconic subviewbutton-nav"
                        label="&places.library.title;"
                        closemenu="none"
-                       oncommand="PanelUI.showSubView('appMenu-libraryView', this)"/>
+                       oncommand="PanelUI.showSubView('appMenu-libraryView', this)"
+                       hidden="true"/>
         <toolbarbutton id="appMenu-logins-button"
                        class="subviewbutton subviewbutton-iconic"
                        label="&logins.label;"
                        oncommand="LoginHelper.openPasswordManager(window, { entryPoint: 'mainmenu' })"
-                       />
+                       hidden="true"/>
         <toolbarbutton id="appMenu-addons-button"
                        class="subviewbutton subviewbutton-iconic"
                        label="&addons.label;"
                        key="key_openAddons"
                        command="Tools:Addons"
-                       />
+                       hidden="true"/>
+        <toolbarbutton id="appMenu-library-history-button"
+                       class="subviewbutton subviewbutton-iconic subviewbutton-nav"
+                       label="&historyMenu.label;"
+                       closemenu="none"
+                       oncommand="PanelUI.showSubView('PanelUI-history', this)"/>
+        <toolbarbutton id="appMenu-library-bookmarks-button"
+                       class="subviewbutton subviewbutton-iconic subviewbutton-nav"
+                       data-l10n-id="library-bookmarks-menu"
+                       closemenu="none"
+                       oncommand="BookmarkingUI.showSubView(this);"/>
+        <toolbarbutton id="appMenu-library-downloads-button"
+                       class="subviewbutton subviewbutton-iconic subviewbutton-nav"
+                       label="&libraryDownloads.label;"
+                       closemenu="none"
+                       oncommand="DownloadsSubview.show(this);"/>
         <toolbarbutton id="appMenu-preferences-button"
                        class="subviewbutton subviewbutton-iconic"
 #ifdef XP_WIN
@@ -879,8 +901,7 @@
                        key="key_preferencesCmdMac"
 #endif
 #endif
-                       oncommand="openPreferences()"
-                       />
+                       oncommand="openPreferences()"/>
         <toolbarbutton id="appMenu-customize-button"
                        class="subviewbutton subviewbutton-iconic"
                        data-l10n-id="appmenu-customizetoolbar"
@@ -910,11 +931,13 @@
 #endif
                        />
         <toolbarseparator/>
+
         <toolbarbutton id="appMenu-find-button"
                        class="subviewbutton subviewbutton-iconic"
                        label="&findOnCmd.label;"
                        key="key_find"
                        command="cmd_find"/>
+        <toolbarseparator/>
         <toolbarbutton id="appMenu-more-button"
                        class="subviewbutton subviewbutton-nav"
                        label="&moreMenu.label;"
