diff --git a/browser/components/about/AboutRedirector.cpp b/browser/components/about/AboutRedirector.cpp
index e3ff9023ca9f02d7f8d53e93333fe25ce4057038..96b3f0375211256a01c779e933cfd41e81983e9e 100644
--- a/browser/components/about/AboutRedirector.cpp
+++ b/browser/components/about/AboutRedirector.cpp
@@ -127,7 +127,7 @@ static const RedirEntry kRedirMap[] = {
          nsIAboutModule::URI_MUST_LOAD_IN_CHILD | nsIAboutModule::ALLOW_SCRIPT |
          nsIAboutModule::URI_CAN_LOAD_IN_PRIVILEGEDABOUT_PROCESS},
     {"ion", "chrome://browser/content/ion.html",
-     nsIAboutModule::ALLOW_SCRIPT | nsIAboutModule::HIDE_FROM_ABOUTABOUT},
+     nsIAboutModule::ALLOW_SCRIPT | nsIAboutModule::HIDE_FROM_ABOUTABOUT}
 };
 
 static nsAutoCString GetAboutModuleName(nsIURI* aURI) {
@@ -197,6 +197,13 @@ AboutRedirector::NewChannel(nsIURI* aURI, nsILoadInfo* aLoadInfo,
         NS_ENSURE_SUCCESS(rv, rv);
       }
 
+      if (
+        path.EqualsLiteral("preferences") && 
+        Preferences::GetBool("browser.preferences.webuiexperience.enabled", true)
+      ) {
+        url.AssignASCII("chrome://browser/content/preferences/new/newpreferences.html");
+      }
+
       // fall back to the specified url in the map
       if (url.IsEmpty()) {
         url.AssignASCII(redir.url);
