diff --git a/browser/base/jar.mn b/browser/base/jar.mn
index 6f825da44aa9ae98523fda0d9f88236373fe1965..69d60b338e6b463945d40a0d9569c305df6bc06e 100644
--- a/browser/base/jar.mn
+++ b/browser/base/jar.mn
@@ -77,11 +77,6 @@ browser.jar:
         content/browser/defaultthemes/5.icon.jpg            (content/defaultthemes/5.icon.jpg)
         content/browser/defaultthemes/5.preview.jpg         (content/defaultthemes/5.preview.jpg)
         content/browser/history-swipe-arrow.svg             (content/history-swipe-arrow.svg)
-*       content/browser/pageinfo/pageInfo.xhtml             (content/pageinfo/pageInfo.xhtml)
-        content/browser/pageinfo/pageInfo.js                (content/pageinfo/pageInfo.js)
-        content/browser/pageinfo/pageInfo.css               (content/pageinfo/pageInfo.css)
-        content/browser/pageinfo/permissions.js             (content/pageinfo/permissions.js)
-        content/browser/pageinfo/security.js                (content/pageinfo/security.js)
         content/browser/robot.ico                           (content/robot.ico)
         content/browser/static-robot.png                    (content/static-robot.png)
         content/browser/safeMode.css                        (content/safeMode.css)
