diff --git a/browser/extensions/moz.build b/browser/extensions/moz.build
index 269dcb2a32d6873e9b697f4c35b9180dcc4daaf9..e8f56b64dfbf34907a9f39149d73956926c676e9 100644
--- a/browser/extensions/moz.build
+++ b/browser/extensions/moz.build
@@ -7,8 +7,10 @@
 DIRS += [
     "doh-rollout",
     "formautofill",
+    "ntp", "qrcodes",
     "screenshots",
     "webcompat",
+    "shield",
     "report-site-issue",
     "pictureinpicture",
 ]
