diff --git a/browser/components/about/components.conf b/browser/components/about/components.conf
index 274eee9c384ee5aaf3b6e23fb266ba4908c4c0d6..ef5e3bb087ef39ffc201e888454d59adf56fd3d2 100644
--- a/browser/components/about/components.conf
+++ b/browser/components/about/components.conf
@@ -9,6 +9,7 @@ pages = [
     'certerror',
     'downloads',
     'framecrashed',
+    'newinstall',
     'home',
     'logins',
     'loginsimportreport',
@@ -18,6 +19,7 @@ pages = [
     'pocket-saved',
     'pocket-signup',
     'policies',
+    'newpreferences',
     'preferences',
     'privatebrowsing',
     'protections',
