diff --git a/browser/components/moz.build b/browser/components/moz.build
index 78fce1bfb4641d1b0fe6857fe593db12f4ee8e60..b46baf48d9e1d1d0a87cc6060fac88daf9dce634 100644
--- a/browser/components/moz.build
+++ b/browser/components/moz.build
@@ -29,6 +29,7 @@ with Files("controlcenter/**"):
 DIRS += [
     "about",
     "aboutconfig",
+    "abouthistory",
     "aboutlogins",
     "attribution",
     "contextualidentity",
