diff --git a/browser/components/moz.build b/browser/components/moz.build
index ba617de9d82f2e165b50411c31acd90c71243636..994d1fedd655c8e47b477257f1b74d63ef14113b 100644
--- a/browser/components/moz.build
+++ b/browser/components/moz.build
@@ -29,6 +29,7 @@ with Files("controlcenter/**"):
 DIRS += [
     "about",
     "aboutconfig",
+    "abouthistory",
     "aboutlogins",
     "attribution",
     "contextualidentity",
