diff --git a/browser/themes/shared/jar.inc.mn b/browser/themes/shared/jar.inc.mn
index 9a62864b3f1ff80db532019949f6c480175de35f..219711a0186555a12d8de920ec5e3312e14a4972 100644
--- a/browser/themes/shared/jar.inc.mn
+++ b/browser/themes/shared/jar.inc.mn
@@ -71,6 +71,8 @@
   skin/classic/browser/notification-icons/drag-indicator.svg                (../shared/notification-icons/drag-indicator.svg)
   skin/classic/browser/notification-icons/focus-tab-by-prompt.svg           (../shared/notification-icons/focus-tab-by-prompt.svg)
   skin/classic/browser/notification-icons/indexedDB.svg                     (../shared/notification-icons/indexedDB.svg)
+  skin/classic/browser/notification-icons/javascript.svg                    (../shared/notification-icons/javascript.svg)
+  skin/classic/browser/notification-icons/location.svg                      (../shared/notification-icons/location.svg)
   skin/classic/browser/notification-icons/microphone-blocked.svg            (../shared/notification-icons/microphone-blocked.svg)
   skin/classic/browser/notification-icons/microphone-detailed.svg           (../shared/notification-icons/microphone-detailed.svg)
   skin/classic/browser/notification-icons/microphone.svg                    (../shared/notification-icons/microphone.svg)
@@ -148,8 +150,10 @@
   skin/classic/browser/bookmark-hollow.svg            (../shared/icons/bookmark-hollow.svg)
   skin/classic/browser/bookmark-star-on-tray.svg      (../shared/icons/bookmark-star-on-tray.svg)
   skin/classic/browser/bookmarks-toolbar.svg          (../shared/icons/bookmarks-toolbar.svg)
+  skin/classic/browser/certificate.svg                (../shared/icons/certificate.svg)
   skin/classic/browser/characterEncoding.svg          (../shared/icons/characterEncoding.svg)
   skin/classic/browser/chevron-animation.svg          (../shared/icons/chevron-animation.svg)
+  skin/classic/browser/cookies.svg                    (../shared/icons/cookies.svg)
   skin/classic/browser/customize.svg                  (../shared/icons/customize.svg)
   skin/classic/browser/developer.svg                  (../shared/icons/developer.svg)
   skin/classic/browser/device-phone.svg               (../shared/icons/device-phone.svg)
@@ -190,6 +194,7 @@
   skin/classic/browser/privateBrowsing.svg            (../shared/icons/privateBrowsing.svg)
   skin/classic/browser/reader-mode.svg                (../shared/icons/reader-mode.svg)
   skin/classic/browser/restore-session.svg            (../shared/icons/restore-session.svg)
+  skin/classic/browser/qr-code.svg                    (../shared/icons/qr-code.svg)
   skin/classic/browser/quit.svg                       (../shared/icons/quit.svg)
   skin/classic/browser/reload.svg                     (../shared/icons/reload.svg)
   skin/classic/browser/reload-to-stop.svg             (../shared/icons/reload-to-stop.svg)
@@ -203,6 +208,7 @@
 #ifndef MOZ_WIDGET_GTK
   skin/classic/browser/sort.svg                       (../shared/icons/sort.svg)
 #endif
+  skin/classic/browser/spell-check.svg                (../shared/icons/spell-check.svg)
   skin/classic/browser/stop.svg                       (../shared/icons/stop.svg)
   skin/classic/browser/stop-to-reload.svg             (../shared/icons/stop-to-reload.svg)
   skin/classic/browser/success-animation.svg          (../shared/icons/success-animation.svg)
