diff --git a/browser/components/BrowserGlue.jsm b/browser/components/BrowserGlue.jsm
index 869a90ef40f99846fa570f53b094e9bb688c8c34..1c286fff06ef50da9695a133124bb59ae2b2411f 100644
--- a/browser/components/BrowserGlue.jsm
+++ b/browser/components/BrowserGlue.jsm
@@ -1408,16 +1408,16 @@ BrowserGlue.prototype = {
       "resource://builtin-themes/dark/"
     );
     AddonManager.maybeInstallBuiltinAddon(
-      "firefox-alpenglow@mozilla.org",
-      "1.2",
-      "resource://builtin-themes/alpenglow/"
+      "dot-fusion@dothq.co",
+      "0.1",
+      "resource://builtin-themes/fusion/"
     );
 
     if (AppConstants.MOZ_NORMANDY) {
       Normandy.init();
     }
 
-    SaveToPocket.init();
+    // SaveToPocket.init();
 
     AboutHomeStartupCache.init();
 
