diff --git a/browser/tools/mozscreenshots/mozscreenshots/extension/configurations/ControlCenter.jsm b/browser/tools/mozscreenshots/mozscreenshots/extension/configurations/ControlCenter.jsm
index 7caca140733dcb61c4cf4a7c3e536baafe9cdc06..17c959d87baf10a9e1a49e279c07d886797b8e01 100644
--- a/browser/tools/mozscreenshots/mozscreenshots/extension/configurations/ControlCenter.jsm
+++ b/browser/tools/mozscreenshots/mozscreenshots/extension/configurations/ControlCenter.jsm
@@ -314,7 +314,7 @@ async function openIdentityPopup(expand) {
         .click();
     } else {
       gIdentityHandler._identityPopup
-        .querySelector("#identity-popup-security-expander")
+        .querySelector("#identity-popup-certificate")
         .click();
     }
   }
