diff --git a/browser/components/extensions/parent/ext-sidebarAction.js b/browser/components/extensions/parent/ext-sidebarAction.js
index b30128661e3f5c56d1db82c60d4919a5dfedef77..cdceb531e22aa30e0b97d465a4758bead291f060 100644
--- a/browser/components/extensions/parent/ext-sidebarAction.js
+++ b/browser/components/extensions/parent/ext-sidebarAction.js
@@ -185,19 +185,19 @@ this.sidebarAction = class extends ExtensionAPI {
     // Insert a toolbarbutton for the sidebar dropdown selector.
     let toolbarbutton = document.createXULElement("toolbarbutton");
     toolbarbutton.setAttribute("id", this.buttonId);
-    toolbarbutton.setAttribute("type", "checkbox");
+    toolbarbutton.setAttribute("type", "radio");
     toolbarbutton.setAttribute("label", details.title);
-    toolbarbutton.setAttribute("oncommand", `SidebarUI.show("${this.id}");`);
+    toolbarbutton.setAttribute("oncommand", `SidebarUI.toggle("${this.id}");`);
     toolbarbutton.setAttribute(
       "class",
-      "subviewbutton subviewbutton-iconic webextension-menuitem"
+      "sidebar-iconbar-item-ext sidebar-iconbar-item"
     );
     toolbarbutton.setAttribute("key", keyId);
     this.setMenuIcon(toolbarbutton, details);
 
     document.getElementById("viewSidebarMenu").appendChild(menuitem);
-    let separator = document.getElementById("sidebar-extensions-separator");
-    separator.parentNode.insertBefore(toolbarbutton, separator);
+    let iconBar = document.getElementById("sidebar-iconbar");
+    iconBar.childNodes[0].appendChild(toolbarbutton);
     SidebarUI.updateShortcut({ button: toolbarbutton });
 
     return menuitem;
