diff --git a/browser/base/content/aboutNetError.xhtml b/browser/base/content/aboutNetError.xhtml
index f612d4df4c0edac5218a69eb50d0d2ce79a16b54..a9cb6c1615f736f248368d1bc5473267ab6f2715 100644
--- a/browser/base/content/aboutNetError.xhtml
+++ b/browser/base/content/aboutNetError.xhtml
@@ -213,4 +213,5 @@
   <script src="chrome://global/content/certviewer/asn1js_bundle.js"></script>
   <script src="chrome://global/content/certviewer/pkijs_bundle.js"></script>
   <script type="module" src="chrome://browser/content/aboutNetError.js"/>
+  <script src="chrome://browser/content/contentTheme.js"/>
 </html>
