diff --git a/browser/components/sessionstore/content/aboutSessionRestore.xhtml b/browser/components/sessionstore/content/aboutSessionRestore.xhtml
index 8e78064f94255cd223787a330a4c6b944a69d38f..cc96c807aca4e53891f05c95b1a2a566842c7f20 100644
--- a/browser/components/sessionstore/content/aboutSessionRestore.xhtml
+++ b/browser/components/sessionstore/content/aboutSessionRestore.xhtml
@@ -19,6 +19,7 @@
     <link rel="localization" href="browser/aboutSessionRestore.ftl"/>
     <link rel="localization" href="branding/brand.ftl"/>
     <script src="chrome://browser/content/aboutSessionRestore.js"/>
+    <script src="chrome://browser/content/contentTheme.js"></script>
   </head>
 
   <body class="illustrated">
