diff --git a/browser/locales/en-US/chrome/browser/browser.dtd b/browser/locales/en-US/chrome/browser/browser.dtd
index 2fd7c1503cf54479ced001a0032d83b07090b0be..1ea4ae58889afb1e4c85dd48e1a07c49dfa1ae8d 100644
--- a/browser/locales/en-US/chrome/browser/browser.dtd
+++ b/browser/locales/en-US/chrome/browser/browser.dtd
@@ -68,6 +68,7 @@ this container is a toolbar. This avoids double-speaking. -->
 
 <!ENTITY webDeveloperMenu.label       "Web Developer">
 
+<!ENTITY newTabCmd.label        "New Tab">
 <!ENTITY newNavigatorCmd.label        "New Window">
 <!ENTITY newPrivateWindow.label     "New Private Window">
 
