diff --git a/browser/base/content/browser-siteIdentity.js b/browser/base/content/browser-siteIdentity.js
index 5850d7ceb4c2a83cc450e5f2e92c8ac8ac23d3d9..3a833fc106c96e97ed745eb69f298b5bb093090b 100644
--- a/browser/base/content/browser-siteIdentity.js
+++ b/browser/base/content/browser-siteIdentity.js
@@ -57,7 +57,7 @@ var gIdentityHandler = {
    * RegExp used to decide if an about url should be shown as being part of
    * the browser UI.
    */
-  _secureInternalPages: /^(?:accounts|addons|cache|certificate|config|crashes|downloads|license|logins|preferences|protections|rights|sessionrestore|support|welcomeback|ion)(?:[?#]|$)/i,
+  _secureInternalPages: /^(?:accounts|addons|cache|certificate|config|crashes|downloads|license|logins|newpreferences|preferences|protections|rights|sessionrestore|support|welcomeback|ion|history)(?:[?#]|$)/i,
 
   /**
    * Whether the established HTTPS connection is considered "broken".
