diff --git a/browser/components/search/content/searchbar.js b/browser/components/search/content/searchbar.js
index 54c2e97443a8a1836df5492750af809669bbd6e1..21893098e69e00a74fd2f0ef1c538971693a2dc6 100644
--- a/browser/components/search/content/searchbar.js
+++ b/browser/components/search/content/searchbar.js
@@ -814,14 +814,14 @@
 
     _buildContextMenu() {
       const raw = `
-        <menuitem data-l10n-id="text-action-undo" cmd="cmd_undo"/>
+        <menuitem class="menuitem-iconic" image="chrome://global/skin/icons/undo.svg" data-l10n-id="text-action-undo" cmd="cmd_undo"/>
         <menuitem data-l10n-id="text-action-redo" cmd="cmd_redo"/>
         <menuseparator/>
-        <menuitem data-l10n-id="text-action-cut" cmd="cmd_cut"/>
-        <menuitem data-l10n-id="text-action-copy" cmd="cmd_copy"/>
-        <menuitem data-l10n-id="text-action-paste" cmd="cmd_paste"/>
-        <menuitem class="searchbar-paste-and-search"/>
-        <menuitem data-l10n-id="text-action-delete" cmd="cmd_delete"/>
+         <menuitem class="menuitem-iconic" image="chrome://browser/skin/edit-cut.svg" data-l10n-id="text-action-cut" cmd="cmd_cut"/>
+        <menuitem class="menuitem-iconic" image="chrome://browser/skin/edit-copy.svg" data-l10n-id="text-action-copy" cmd="cmd_copy"/>
+        <menuitem class="menuitem-iconic" image="chrome://browser/skin/edit-paste.svg" data-l10n-id="text-action-paste" cmd="cmd_paste"/>
+        <menuitem class="menuitem-iconic" image="chrome://global/skin/icons/search-glass.svg" class="searchbar-paste-and-search"/>
+        <menuitem class="menuitem-iconic" image="chrome://global/skin/icons/delete.svg" data-l10n-id="text-action-delete" cmd="cmd_delete"/>
         <menuitem data-l10n-id="text-action-select-all" cmd="cmd_selectAll"/>
         <menuseparator/>
         <menuitem class="searchbar-clear-history"/>
