diff --git a/browser/base/jar.mn b/browser/base/jar.mn
index 8d482e8b6417223807cc618e8f31728698501e15..03a7ab52e9c0879ffd402a02a641c79370f9365b 100644
--- a/browser/base/jar.mn
+++ b/browser/base/jar.mn
@@ -79,11 +79,6 @@ browser.jar:
         content/browser/defaultthemes/5.icon.jpg            (content/defaultthemes/5.icon.jpg)
         content/browser/defaultthemes/5.preview.jpg         (content/defaultthemes/5.preview.jpg)
         content/browser/history-swipe-arrow.svg             (content/history-swipe-arrow.svg)
-*       content/browser/pageinfo/pageInfo.xhtml             (content/pageinfo/pageInfo.xhtml)
-        content/browser/pageinfo/pageInfo.js                (content/pageinfo/pageInfo.js)
-        content/browser/pageinfo/pageInfo.css               (content/pageinfo/pageInfo.css)
-        content/browser/pageinfo/permissions.js             (content/pageinfo/permissions.js)
-        content/browser/pageinfo/security.js                (content/pageinfo/security.js)
         content/browser/robot.ico                           (content/robot.ico)
         content/browser/static-robot.png                    (content/static-robot.png)
         content/browser/safeMode.css                        (content/safeMode.css)
