diff --git a/browser/modules/ThemeVariableMap.jsm b/browser/modules/ThemeVariableMap.jsm
index 8178dd9a8da5e43383d633e25ad5643dfd07e5e3..e857e81b63b6f4e0cf89a6a36433bcd801396f18 100644
--- a/browser/modules/ThemeVariableMap.jsm
+++ b/browser/modules/ThemeVariableMap.jsm
@@ -181,6 +181,13 @@ const ThemeVariableMap = [
       optionalElementID: "browser",
     },
   ],
+  [
+    "--sidebar-highlight-color",
+    {
+      lwtProperty: "sidebar_highlight",
+      optionalElementID: "browser",
+    },
+  ],
 ];
 
 const ThemeContentPropertyList = [
