diff --git a/browser/base/content/browser-pageActions.js b/browser/base/content/browser-pageActions.js
index 16ff6eae1edebd418c272e2f6a2109971d8a7d80..af25313771d9e67b259de4bc88ca071eaf1ccf42 100644
--- a/browser/base/content/browser-pageActions.js
+++ b/browser/base/content/browser-pageActions.js
@@ -1355,7 +1355,7 @@ BrowserPageActions.shareURL = {
       PanelMultiView.hidePopup(BrowserPageActions.panelNode);
     };
 
-    shareProviders.forEach(function(share) {
+    shareProviders.forEach(function (share) {
       let item = document.createXULElement("toolbarbutton");
       item.setAttribute("label", share.menuItemTitle);
       item.setAttribute("share-name", share.name);
@@ -1383,6 +1383,26 @@ BrowserPageActions.shareURL = {
   },
 };
 
+// generate QR code for site
+BrowserPageActions.generateQrCode = {
+  get panel() {
+    return document.getElementById("qrCodePanel")
+  },
+
+  onShowingInPanel(buttonNode) {
+    console.log("showing in panel")
+  },
+
+  onCommand(event, buttonNode) {
+    PanelMultiView.hidePopup(BrowserPageActions.panelNode);
+    this.panel.openPopupAtScreen(
+      200,
+      100,
+      false
+    );
+  },
+};
+
 // Attach sharingService here so tests can override the implementation
 XPCOMUtils.defineLazyServiceGetters(BrowserPageActions.shareURL, {
   _sharingService: [
