diff --git a/browser/components/newtab/aboutwelcome/AboutWelcomeParent.jsm b/browser/components/newtab/aboutwelcome/AboutWelcomeParent.jsm
index e963272ca8ed9802240dfca680114384fd94c6fc..af3f5a052028bd3e651d9e30c5ec737a6732377f 100644
--- a/browser/components/newtab/aboutwelcome/AboutWelcomeParent.jsm
+++ b/browser/components/newtab/aboutwelcome/AboutWelcomeParent.jsm
@@ -48,9 +48,9 @@ const AWTerminate = {
   ADDRESS_BAR_NAVIGATED: "address-bar-navigated",
 };
 const LIGHT_WEIGHT_THEMES = {
-  DARK: "firefox-compact-dark@mozilla.org",
-  LIGHT: "firefox-compact-light@mozilla.org",
-  AUTOMATIC: "default-theme@mozilla.org",
+  DARK: "dark@themes.dothq.co",
+  LIGHT: "light@themes.dothq.co",
+  AUTOMATIC: "dynamic@themes.dothq.co",
   ALPENGLOW: "firefox-alpenglow@mozilla.org",
 };
 
