diff --git a/browser/themes/addons/jar.mn b/browser/themes/addons/jar.mn
index 5b131fba5ef13ca54e659e1e366b44e1d74785a3..7566659326e1b84883389534bb27e0c493dfd318 100644
--- a/browser/themes/addons/jar.mn
+++ b/browser/themes/addons/jar.mn
@@ -5,8 +5,8 @@
 browser.jar:
 %  resource builtin-themes %content/builtin-themes/
 
-  content/builtin-themes/alpenglow                 (alpenglow/*.svg)
-  content/builtin-themes/alpenglow/manifest.json   (alpenglow/manifest.json)
+  content/builtin-themes/fusion                    (fusion/*.svg)
+  content/builtin-themes/fusion/manifest.json      (fusion/manifest.json)
 
   content/builtin-themes/dark                      (dark/*.svg)
   content/builtin-themes/dark                      (dark/*.css)
