diff --git a/browser/base/content/browser-menubar.inc b/browser/base/content/browser-menubar.inc
index 058c1e5835e0a9dd3e92952cd8b987fd06d00ed7..f4dee7bd72a9d0ed64a8a80bdb8c7683ac793d0c 100644
--- a/browser/base/content/browser-menubar.inc
+++ b/browser/base/content/browser-menubar.inc
@@ -96,23 +96,33 @@
                          onpopuphidden="updateEditUIVisibility()">
                 <menuitem id="menu_undo"
                           key="key_undo"
-                          command="cmd_undo" data-l10n-id="text-action-undo"/>
+                          command="cmd_undo" data-l10n-id="text-action-undo"
+                          class="menuitem-iconic"
+                          image="chrome://browser/skin/undo.svg"/>
                 <menuitem id="menu_redo"
                           key="key_redo"
                           command="cmd_redo" data-l10n-id="text-action-redo"/>
                 <menuseparator/>
                 <menuitem id="menu_cut"
                           key="key_cut"
-                          command="cmd_cut" data-l10n-id="text-action-cut"/>
+                          command="cmd_cut" data-l10n-id="text-action-cut"
+                          class="menuitem-iconic"
+                          image="chrome://browser/skin/edit-cut.svg"/>
                 <menuitem id="menu_copy"
                           key="key_copy"
-                          command="cmd_copy" data-l10n-id="text-action-copy"/>
+                          command="cmd_copy" data-l10n-id="text-action-copy"
+                          class="menuitem-iconic"
+                          image="chrome://browser/skin/edit-copy.svg"/>
                 <menuitem id="menu_paste"
                           key="key_paste"
-                          command="cmd_paste" data-l10n-id="text-action-paste"/>
+                          command="cmd_paste" data-l10n-id="text-action-paste"
+                          class="menuitem-iconic"
+                          image="chrome://browser/skin/edit-paste.svg"/>
                 <menuitem id="menu_delete"
                           key="key_delete"
-                          command="cmd_delete" data-l10n-id="text-action-delete"/>
+                          command="cmd_delete" data-l10n-id="text-action-delete"
+                          class="menuitem-iconic"
+                          image="chrome://browser/skin/icons/delete.svg"/>
                 <menuseparator/>
                 <menuitem id="menu_selectAll"
                           key="key_selectAll"
@@ -149,17 +159,20 @@
                 <menu id="viewSidebarMenuMenu" data-l10n-id="menu-view-sidebar">
                   <menupopup id="viewSidebarMenu">
                     <menuitem id="menu_bookmarksSidebar"
-                              type="checkbox"
+                              type="radio"
                               key="viewBookmarksSidebarKb"
                               oncommand="SidebarUI.toggle('viewBookmarksSidebar');" data-l10n-id="menu-view-bookmarks"/>
                     <menuitem id="menu_historySidebar"
-                              type="checkbox"
+                              type="radio"
                               key="key_gotoHistory"
                               oncommand="SidebarUI.toggle('viewHistorySidebar');" data-l10n-id="menu-view-history-button"/>
                     <menuitem id="menu_tabsSidebar"
-                              type="checkbox"
+                              type="radio"
                               class="sync-ui-item"
                               oncommand="SidebarUI.toggle('viewTabsSidebar');" data-l10n-id="menu-view-synced-tabs-sidebar"/>
+                    <menuitem id="menu_accountsSidebar"
+                              type="radio"
+                              oncommand="SidebarUI.toggle('viewAccountsSidebar');" data-l10n-id="menu-view-accounts-button"/>
                   </menupopup>
                 </menu>
                 <menuseparator/>
