diff --git a/browser/app/moz.build b/browser/app/moz.build
index 9da2da337663a59ad4bea58733abab897a630415..6297fbbc0eb5f0a41e0bf0554d937d19a19dc6d9 100644
--- a/browser/app/moz.build
+++ b/browser/app/moz.build
@@ -26,6 +26,8 @@ with Files("Makefile.in"):
 
 with Files("profile/channel-prefs.js"):
     BUG_COMPONENT = ("Firefox", "Installer")
+with Files("profile/dot.js"):
+    BUG_COMPONENT = ("Firefox", "General")
 with Files("profile/firefox.js"):
     BUG_COMPONENT = ("Firefox", "General")
 
