diff --git a/browser/actors/moz.build b/browser/actors/moz.build
index 5118f85caf541ad0466db65a2a3010e5abd65873..86e8b9360190cde49d206a9ab2879719a3d42a65 100644
--- a/browser/actors/moz.build
+++ b/browser/actors/moz.build
@@ -29,6 +29,8 @@ with Files("WebRTCChild.jsm"):
     BUG_COMPONENT = ("Firefox", "Site Permissions")
 
 FINAL_TARGET_FILES.actors += [
+    "AboutHistoryChild.jsm",
+    "AboutHistoryParent.jsm",
     "AboutNewInstallChild.jsm",
     "AboutNewInstallParent.jsm",
     "AboutNewTabChild.jsm",
