diff --git a/browser/components/customizableui/CustomizeMode.jsm b/browser/components/customizableui/CustomizeMode.jsm
index 7b97b88e79b1debbf73853fcf2ba6745725c9dba..1e4cb95a95261227969d3a7d05f207285cbbef37 100644
--- a/browser/components/customizableui/CustomizeMode.jsm
+++ b/browser/components/customizableui/CustomizeMode.jsm
@@ -89,16 +89,14 @@ XPCOMUtils.defineLazyGetter(this, "log", () => {
   return new scope.ConsoleAPI(consoleOptions);
 });
 
-const DEFAULT_THEME_ID = "default-theme@mozilla.org";
-const LIGHT_THEME_ID = "firefox-compact-light@mozilla.org";
-const DARK_THEME_ID = "firefox-compact-dark@mozilla.org";
-const ALPENGLOW_THEME_ID = "firefox-alpenglow@mozilla.org";
+const DEFAULT_THEME_ID = "dynamic@themes.dothq.co";
+const LIGHT_THEME_ID = "light@themes.dothq.co";
+const DARK_THEME_ID = "dark@themes.dothq.co";
 
 const _defaultImportantThemes = [
   DEFAULT_THEME_ID,
   LIGHT_THEME_ID,
   DARK_THEME_ID,
-  ALPENGLOW_THEME_ID,
 ];
 
 var gDraggingInToolbars;
