diff --git a/browser/base/content/aboutDialog.js b/browser/base/content/aboutDialog.js
index b96337666cff01dadfb4bbf127ae6e95a86aad98..c4398d47c3bb905722b403a7ac25c4e41999d097 100644
--- a/browser/base/content/aboutDialog.js
+++ b/browser/base/content/aboutDialog.js
@@ -52,7 +52,7 @@ async function init(aEvent) {
   // Include the build ID and display warning if this is an "a#" (nightly or aurora) build
   let versionId = "aboutDialog-version";
   let versionAttributes = {
-    version: AppConstants.MOZ_APP_VERSION_DISPLAY,
+    version: AppConstants.DOT_APP_VERSION || AppConstants.MOZ_APP_VERSION_DISPLAY,
     bits: Services.appinfo.is64Bit ? 64 : 32,
   };
 
