diff --git a/browser/components/preferences/moz.build b/browser/components/preferences/moz.build
index a42f5f0921e8b94225260d30597d6a90606cb255..5df9e7ccf308095710940ae16fcfa98c9dbd8f03 100644
--- a/browser/components/preferences/moz.build
+++ b/browser/components/preferences/moz.build
@@ -4,7 +4,7 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
-DIRS += ["dialogs"]
+DIRS += ["dialogs", "new"]
 
 BROWSER_CHROME_MANIFESTS += ["tests/browser.ini", "tests/siteData/browser.ini"]
 
