diff --git a/browser/components/urlbar/UrlbarInput.jsm b/browser/components/urlbar/UrlbarInput.jsm
index 73d15dff8e5b82c236e8022c71cacd7b0de4bd48..0d4f6a431999c1e04e388b3387715f6b68357c05 100644
--- a/browser/components/urlbar/UrlbarInput.jsm
+++ b/browser/components/urlbar/UrlbarInput.jsm
@@ -2516,6 +2516,8 @@ class UrlbarInput {
       .GetStringFromName("pasteAndGo.label");
     pasteAndGo.setAttribute("label", label);
     pasteAndGo.setAttribute("anonid", "paste-and-go");
+    pasteAndGo.setAttribute("image", "chrome://global/skin/icons/search-glass.svg");
+    pasteAndGo.classList.add("menuitem-iconic");
     pasteAndGo.addEventListener("command", () => {
       this._suppressStartQuery = true;
 
