diff --git a/browser/components/about/components.conf b/browser/components/about/components.conf
index 42011d5094813d8360f3c4e6a8436efb14758a3c..d4aa194a761cd0748aff55261b4902ac60aef4a9 100644
--- a/browser/components/about/components.conf
+++ b/browser/components/about/components.conf
@@ -10,6 +10,7 @@ pages = [
     'config',
     'downloads',
     'framecrashed',
+    'history',
     'home',
     'logins',
     'newinstall',
