diff --git a/browser/components/extensions/parent/ext-sidebarAction.js b/browser/components/extensions/parent/ext-sidebarAction.js
index 99aa63874bc6f1821baae6788fbba18f6861f9c6..fb4b4f92f6b2ff6bb0e78a0795eb90073ff24895 100644
--- a/browser/components/extensions/parent/ext-sidebarAction.js
+++ b/browser/components/extensions/parent/ext-sidebarAction.js
@@ -178,25 +178,26 @@ this.sidebarAction = class extends ExtensionAPI {
     menuitem.setAttribute("type", "checkbox");
     menuitem.setAttribute("label", details.title);
     menuitem.setAttribute("oncommand", `SidebarUI.toggle("${this.id}");`);
-    menuitem.setAttribute("class", "menuitem-iconic webextension-menuitem");
+    menuitem.setAttribute("class", "sidebar-iconbar-item-ext sidebar-iconbar-item");
     menuitem.setAttribute("key", keyId);
     this.setMenuIcon(menuitem, details);
 
     // Insert a toolbarbutton for the sidebar dropdown selector.
     let toolbarbutton = document.createXULElement("toolbarbutton");
     toolbarbutton.setAttribute("id", this.buttonId);
+    toolbarbutton.setAttribute("type", "radio");
     toolbarbutton.setAttribute("label", details.title);
-    toolbarbutton.setAttribute("oncommand", `SidebarUI.show("${this.id}");`);
+    toolbarbutton.setAttribute("oncommand", `SidebarUI.toggle("${this.id}");`);
     toolbarbutton.setAttribute(
       "class",
-      "subviewbutton subviewbutton-iconic webextension-menuitem"
+      "sidebar-iconbar-item-ext sidebar-iconbar-item"
     );
     toolbarbutton.setAttribute("key", keyId);
     this.setMenuIcon(toolbarbutton, details);
 
     document.getElementById("viewSidebarMenu").appendChild(menuitem);
-    let separator = document.getElementById("sidebar-extensions-separator");
-    separator.parentNode.insertBefore(toolbarbutton, separator);
+    let iconBar = document.getElementById("sidebar-iconbar");
+    iconBar.childNodes[0].appendChild(toolbarbutton);
     SidebarUI.updateShortcut({ button: toolbarbutton });
 
     return menuitem;
