diff --git a/browser/locales/en-US/browser/browser.ftl b/browser/locales/en-US/browser/browser.ftl
index 8602d7056a92610df2cd53240da274d97d32b9e7..4db79bfc9318ca76bad19563511e6115a9f1faa6 100644
--- a/browser/locales/en-US/browser/browser.ftl
+++ b/browser/locales/en-US/browser/browser.ftl
@@ -183,6 +183,9 @@ page-action-send-tabs-urlbar =
       [1] Send Tab to Device
      *[other] Send { $tabCount } Tabs to Device
   }
+
+page-action-qr-code-panel =
+  .label = Generate QR Code for Page
 page-action-pocket-panel =
   .label = Save Page to { -pocket-brand-name }
 page-action-copy-url-panel =
@@ -467,6 +470,8 @@ urlbar-page-action-button =
   .tooltiptext = Page actions
 urlbar-pocket-button =
   .tooltiptext = Save to { -pocket-brand-name }
+urlbar-qr-code-button =
+  .tooltiptext = Generate QR Code for this Page
 
 ## Action text shown in urlbar results, usually appended after the search
 ## string or the url, like "result value - action text".
