diff --git a/browser/modules/ThemeVariableMap.jsm b/browser/modules/ThemeVariableMap.jsm
index 32e034aedfa91d0bdc2e77c3136e0ad6fe06729d..66c0cfb94294c6e2f16744f8bb8a9b1bb5539d29 100644
--- a/browser/modules/ThemeVariableMap.jsm
+++ b/browser/modules/ThemeVariableMap.jsm
@@ -173,6 +173,13 @@ const ThemeVariableMap = [
       optionalElementID: "browser",
     },
   ],
+  [
+    "--sidebar-highlight-color",
+    {
+      lwtProperty: "sidebar_highlight",
+      optionalElementID: "browser",
+    },
+  ],
 ];
 
 const ThemeContentPropertyList = [
