diff --git a/browser/base/content/browser-data-submission-info-bar.js b/browser/base/content/browser-data-submission-info-bar.js
index e94a0ef546dbaa592ce475c55f09a148c913c038..32464fa950324ae71b59247e64159081cb381f22 100644
--- a/browser/base/content/browser-data-submission-info-bar.js
+++ b/browser/base/content/browser-data-submission-info-bar.js
@@ -106,7 +106,8 @@ var gDataNotificationInfoBar = {
       case "datareporting:notify-data-policy:request":
         let request = subject.wrappedJSObject.object;
         try {
-          this._displayDataPolicyInfoBar(request);
+          // We don't collect data automatically like the banner says so we have no reason to run this.
+          // this._displayDataPolicyInfoBar(request);
         } catch (ex) {
           request.onUserNotifyFailed(ex);
         }
