diff --git a/browser/base/content/aboutNetError.xhtml b/browser/base/content/aboutNetError.xhtml
index bffb6bea4e29307532eb0071d6570f73b2a763c2..8f26d3a6bdc37694759b0929e55b2cc78f4b0fb0 100644
--- a/browser/base/content/aboutNetError.xhtml
+++ b/browser/base/content/aboutNetError.xhtml
@@ -217,4 +217,5 @@
   </body>
   <script src="chrome://browser/content/aboutNetErrorCodes.js"/>
   <script src="chrome://browser/content/aboutNetError.js"/>
+  <script src="chrome://browser/content/contentTheme.js"/>
 </html>
