diff --git a/browser/components/sessionstore/content/aboutSessionRestore.xhtml b/browser/components/sessionstore/content/aboutSessionRestore.xhtml
index 05c0e4f732ee16eac6f25f321a4a97ae52c4bf6d..3ba38698bce33fbd293f2f8d73195241c833d623 100644
--- a/browser/components/sessionstore/content/aboutSessionRestore.xhtml
+++ b/browser/components/sessionstore/content/aboutSessionRestore.xhtml
@@ -19,6 +19,7 @@
     <link rel="localization" href="browser/aboutSessionRestore.ftl"/>
     <link rel="localization" href="branding/brand.ftl"/>
     <script src="chrome://browser/content/aboutSessionRestore.js"/>
+    <script src="chrome://browser/content/contentTheme.js"></script>
   </head>
 
   <body class="illustrated">
