diff --git a/browser/base/content/nsContextMenu.js b/browser/base/content/nsContextMenu.js
index abd548073026885bf21c18479dc5fa5338c70547..af9c36798c6168376abf4082821eb7de2d2da02d 100644
--- a/browser/base/content/nsContextMenu.js
+++ b/browser/base/content/nsContextMenu.js
@@ -159,7 +159,7 @@ class nsContextMenu {
         "intl.ellipsis",
         Ci.nsIPrefLocalizedString
       ).data;
-    } catch (e) {}
+    } catch (e) { }
 
     // Reset after "on-build-contextmenu" notification in case selection was
     // changed during the notification.
@@ -361,7 +361,7 @@ class nsContextMenu {
         !mailtoHandler.alwaysAskBeforeHandling &&
         mailtoHandler.preferredAction == Ci.nsIHandlerInfo.useHelperApp &&
         mailtoHandler.preferredApplicationHandler instanceof
-          Ci.nsIWebHandlerApp;
+        Ci.nsIWebHandlerApp;
     }
 
     if (
@@ -373,7 +373,7 @@ class nsContextMenu {
       this.linkURL = this.selectionInfo.linkURL;
       try {
         this.linkURI = makeURI(this.linkURL);
-      } catch (ex) {}
+      } catch (ex) { }
 
       this.linkTextStr = this.selectionInfo.linkText;
       this.onPlainTextLink = true;
@@ -631,15 +631,15 @@ class nsContextMenu {
     this.showItem(
       "context-viewpartialsource-selection",
       !this.inAboutDevtoolsToolbox &&
-        this.isContentSelected &&
-        this.selectionInfo.isDocumentLevelSelection
+      this.isContentSelected &&
+      this.selectionInfo.isDocumentLevelSelection
     );
 
     this.showItem(
       "context-print-selection",
       !this.inAboutDevtoolsToolbox &&
-        this.isContentSelected &&
-        this.selectionInfo.isDocumentLevelSelection
+      this.isContentSelected &&
+      this.selectionInfo.isDocumentLevelSelection
     );
 
     var shouldShow = !(
@@ -679,6 +679,15 @@ class nsContextMenu {
       this.onVideo && (!this.inSyntheticDoc || this.inFrame)
     );
     this.setItemAttr("context-viewvideo", "disabled", !this.mediaURL);
+
+    this.showItem(
+      "context-generateqrcode",
+      shouldShow &&
+      this.inTabBrowser &&
+      !this.inFrame &&
+      !this.contentData.documentURIObject.schemeIs("about") &&
+      Services.prefs.getBoolPref("browser.qrcodes.enabled", true)
+    );
   }
 
   initMiscItems() {
@@ -2211,6 +2220,14 @@ class nsContextMenu {
   doCustomCommand(generatedItemId, handlingUserInput) {
     this.actor.doCustomCommand(generatedItemId, handlingUserInput);
   }
+
+  generateQrCode(aEvent) {
+    const action = PageActions.actionForID('qr_dothq_co');
+
+    if (action) {
+      BrowserPageActions.doCommandForAction(action, aEvent, this);
+    }
+  }
 }
 
 XPCOMUtils.defineLazyModuleGetters(nsContextMenu, {
