diff --git a/browser/components/BrowserGlue.jsm b/browser/components/BrowserGlue.jsm
index 81de0fb201173b6661798ee21461eb3800059580..8da7ef55f864307806882cca4789d11cd46d6357 100644
--- a/browser/components/BrowserGlue.jsm
+++ b/browser/components/BrowserGlue.jsm
@@ -179,6 +179,21 @@ let JSPROCESSACTORS = {
  * available at https://firefox-source-docs.mozilla.org/dom/ipc/jsactors.html
  */
 let JSWINDOWACTORS = {
+  AboutHistory: {
+    parent: {
+      moduleURI: "resource:///actors/AboutHistoryParent.jsm",
+    },
+    child: {
+      moduleURI: "resource:///actors/AboutHistoryChild.jsm",
+
+      events: {
+        DOMWindowCreated: { capture: true },
+      },
+    },
+
+    matches: ["about:history"],
+  },
+
   AboutLogins: {
     parent: {
       moduleURI: "resource:///actors/AboutLoginsParent.jsm",
@@ -519,14 +534,17 @@ let JSWINDOWACTORS = {
       moduleURI: "resource:///actors/LightweightThemeChild.jsm",
       events: {
         pageshow: { mozSystemGroup: true },
+        DOMWindowCreated: {},
+        click: {},
       },
     },
     includeChrome: true,
     allFrames: true,
+    remoteTypes: ["privilegedabout"],
     matches: [
-      "about:home",
-      "about:newtab",
-      "about:welcome",
+      "about:*",
+      "about:certerror?*", 
+      "about:neterror?*",
       "chrome://browser/content/syncedtabs/sidebar.xhtml",
       "chrome://browser/content/places/historySidebar.xhtml",
       "chrome://browser/content/places/bookmarksSidebar.xhtml",
@@ -763,8 +781,8 @@ let JSWINDOWACTORS = {
   if (
     Services.prefs.getCharPref(
       "extensions.activeThemeID",
-      "default-theme@mozilla.org"
-    ) != "default-theme@mozilla.org"
+      "dynamic@themes.dothq.co"
+    ) != "dynamic@themes.dothq.co"
   ) {
     return;
   }
@@ -1356,26 +1374,26 @@ BrowserGlue.prototype = {
     SessionStore.init();
 
     AddonManager.maybeInstallBuiltinAddon(
-      "firefox-compact-light@mozilla.org",
+      "light@themes.dothq.co",
       "1.1",
       "resource://builtin-themes/light/"
     );
     AddonManager.maybeInstallBuiltinAddon(
-      "firefox-compact-dark@mozilla.org",
+      "dark@themes.dothq.co",
       "1.1",
       "resource://builtin-themes/dark/"
     );
     AddonManager.maybeInstallBuiltinAddon(
-      "firefox-alpenglow@mozilla.org",
-      "1.2",
-      "resource://builtin-themes/alpenglow/"
+      "fusion@themes.dothq.co",
+      "0.1",
+      "resource://builtin-themes/fusion/"
     );
 
     if (AppConstants.MOZ_NORMANDY) {
       Normandy.init();
     }
 
-    SaveToPocket.init();
+    // SaveToPocket.init();
 
     AboutHomeStartupCache.init();
 
