diff --git a/browser/tools/mozscreenshots/mozscreenshots/extension/configurations/LightweightThemes.jsm b/browser/tools/mozscreenshots/mozscreenshots/extension/configurations/LightweightThemes.jsm
index bb01f0b444068fe1e4385259b308cedbf0b4eeab..0fd42f5b0e85f0b6de931ea04725ffa5af6ef295 100644
--- a/browser/tools/mozscreenshots/mozscreenshots/extension/configurations/LightweightThemes.jsm
+++ b/browser/tools/mozscreenshots/mozscreenshots/extension/configurations/LightweightThemes.jsm
@@ -18,7 +18,7 @@ var LightweightThemes = {
       selectors: [],
       async applyConfig() {
         let addon = await AddonManager.getAddonByID(
-          "default-theme@mozilla.org"
+          "dynamic@themes.dothq.co"
         );
         await addon.enable();
       },
@@ -28,7 +28,7 @@ var LightweightThemes = {
       selectors: [],
       async applyConfig() {
         let addon = await AddonManager.getAddonByID(
-          "firefox-compact-light@mozilla.org"
+          "light@themes.dothq.co"
         );
         await addon.enable();
       },
@@ -38,7 +38,7 @@ var LightweightThemes = {
       selectors: [],
       async applyConfig() {
         let addon = await AddonManager.getAddonByID(
-          "firefox-compact-dark@mozilla.org"
+          "dark@themes.dothq.co"
         );
         await addon.enable();
       },
