diff --git a/browser/base/content/nsContextMenu.js b/browser/base/content/nsContextMenu.js
index 478646df36b38fd505da5e7e74383a1eac46465a..664c0099d8a54c095fa612390f087dad8ce416a8 100644
--- a/browser/base/content/nsContextMenu.js
+++ b/browser/base/content/nsContextMenu.js
@@ -160,7 +160,7 @@ class nsContextMenu {
         "intl.ellipsis",
         Ci.nsIPrefLocalizedString
       ).data;
-    } catch (e) {}
+    } catch (e) { }
 
     // Reset after "on-build-contextmenu" notification in case selection was
     // changed during the notification.
@@ -363,7 +363,7 @@ class nsContextMenu {
         !mailtoHandler.alwaysAskBeforeHandling &&
         mailtoHandler.preferredAction == Ci.nsIHandlerInfo.useHelperApp &&
         mailtoHandler.preferredApplicationHandler instanceof
-          Ci.nsIWebHandlerApp;
+        Ci.nsIWebHandlerApp;
     }
 
     if (
@@ -375,7 +375,7 @@ class nsContextMenu {
       this.linkURL = this.selectionInfo.linkURL;
       try {
         this.linkURI = makeURI(this.linkURL);
-      } catch (ex) {}
+      } catch (ex) { }
 
       this.linkTextStr = this.selectionInfo.linkText;
       this.onPlainTextLink = true;
@@ -603,15 +603,15 @@ class nsContextMenu {
     this.showItem(
       "context-viewpartialsource-selection",
       !this.inAboutDevtoolsToolbox &&
-        this.isContentSelected &&
-        this.selectionInfo.isDocumentLevelSelection
+      this.isContentSelected &&
+      this.selectionInfo.isDocumentLevelSelection
     );
 
     this.showItem(
       "context-print-selection",
       !this.inAboutDevtoolsToolbox &&
-        this.isContentSelected &&
-        this.selectionInfo.isDocumentLevelSelection
+      this.isContentSelected &&
+      this.selectionInfo.isDocumentLevelSelection
     );
 
     var shouldShow = !(
@@ -651,6 +651,15 @@ class nsContextMenu {
       this.onVideo && (!this.inSyntheticDoc || this.inFrame)
     );
     this.setItemAttr("context-viewvideo", "disabled", !this.mediaURL);
+
+    this.showItem(
+      "context-generateqrcode",
+      shouldShow &&
+      this.inTabBrowser &&
+      !this.inFrame &&
+      !this.contentData.documentURIObject.schemeIs("about") &&
+      Services.prefs.getBoolPref("browser.qrcodes.enabled", true)
+    );
   }
 
   initMiscItems() {
@@ -2191,6 +2200,14 @@ class nsContextMenu {
   doCustomCommand(generatedItemId, handlingUserInput) {
     this.actor.doCustomCommand(generatedItemId, handlingUserInput);
   }
+
+  generateQrCode(aEvent) {
+    const action = PageActions.actionForID('qr_dothq_co');
+
+    if (action) {
+      BrowserPageActions.doCommandForAction(action, aEvent, this);
+    }
+  }
 }
 
 XPCOMUtils.defineLazyModuleGetters(nsContextMenu, {
