diff --git a/browser/base/content/browser-pageActions.js b/browser/base/content/browser-pageActions.js
index 096012abde728e5728c5a3de1bc19953921d3659..70f8b1b770545e17b24df8d1513783c7a3019f03 100644
--- a/browser/base/content/browser-pageActions.js
+++ b/browser/base/content/browser-pageActions.js
@@ -1364,7 +1364,7 @@ BrowserPageActions.shareURL = {
       PanelMultiView.hidePopup(BrowserPageActions.panelNode);
     };
 
-    shareProviders.forEach(function(share) {
+    shareProviders.forEach(function (share) {
       let item = document.createXULElement("toolbarbutton");
       item.setAttribute("label", share.menuItemTitle);
       item.setAttribute("share-name", share.name);
@@ -1392,6 +1392,26 @@ BrowserPageActions.shareURL = {
   },
 };
 
+// generate QR code for site
+BrowserPageActions.generateQrCode = {
+  get panel() {
+    return document.getElementById("qrCodePanel")
+  },
+
+  onShowingInPanel(buttonNode) {
+    console.log("showing in panel")
+  },
+
+  onCommand(event, buttonNode) {
+    PanelMultiView.hidePopup(BrowserPageActions.panelNode);
+    this.panel.openPopupAtScreen(
+      200,
+      100,
+      false
+    );
+  },
+};
+
 // Attach sharingService here so tests can override the implementation
 XPCOMUtils.defineLazyServiceGetters(BrowserPageActions.shareURL, {
   _sharingService: [
