diff --git a/browser/components/newtab/content-src/aboutwelcome/aboutwelcome.jsx b/browser/components/newtab/content-src/aboutwelcome/aboutwelcome.jsx
index db2de9ed3f8c51f33cc7dff17a4228d707313be9..b1213f9aed9edf5e9e0c91ce6dd6a2256c31c58e 100644
--- a/browser/components/newtab/content-src/aboutwelcome/aboutwelcome.jsx
+++ b/browser/components/newtab/content-src/aboutwelcome/aboutwelcome.jsx
@@ -30,21 +30,21 @@ class AboutWelcome extends React.PureComponent {
 
     // Record impression with performance data after allowing the page to load
     const recordImpression = domState => {
-      const { domComplete, domInteractive } = performance
-        .getEntriesByType("navigation")
-        .pop();
-      window.AWSendEventTelemetry({
-        event: "IMPRESSION",
-        event_context: {
-          domComplete,
-          domInteractive,
-          mountStart: performance.getEntriesByName("mount").pop().startTime,
-          domState,
-          source: this.props.UTMTerm,
-          page: "about:welcome",
-        },
-        message_id: this.props.messageId,
-      });
+      // const { domComplete, domInteractive } = performance
+      //   .getEntriesByType("navigation")
+      //   .pop();
+      // window.AWSendEventTelemetry({
+      //   event: "IMPRESSION",
+      //   event_context: {
+      //     domComplete,
+      //     domInteractive,
+      //     mountStart: performance.getEntriesByName("mount").pop().startTime,
+      //     domState,
+      //     source: this.props.UTMTerm,
+      //     page: "about:welcome",
+      //   },
+      //   message_id: this.props.messageId,
+      // });
     };
     if (document.readyState === "complete") {
       // Page might have already triggered a load event because it waited for async data,
