diff --git a/browser/components/moz.build b/browser/components/moz.build
index d16b27b1eaa3da7d11117731dfcf0903af71099a..57b7298f6461f967aececa7ffa8ffa64f5e91a07 100644
--- a/browser/components/moz.build
+++ b/browser/components/moz.build
@@ -28,6 +28,7 @@ with Files("controlcenter/**"):
 
 DIRS += [
     "about",
+    "abouthistory",
     "aboutlogins",
     "attribution",
     "contextualidentity",
