diff --git a/build/moz.build b/build/moz.build
index f70a3e2b78dfc96e4d92b1dcbcc237869605b2d4..70afa9a68b21f3ec1520884561134594af91378a 100644
--- a/build/moz.build
+++ b/build/moz.build
@@ -85,7 +85,7 @@ if CONFIG["MOZ_APP_BASENAME"]:
         if CONFIG[var]:
             appini_defines[var] = True
 
-    appini_defines["MOZ_APPUPDATE_HOST"] = "aus5.mozilla.org"
+    appini_defines["MOZ_APPUPDATE_HOST"] = "push.dothq.co"
     if CONFIG["MOZ_APPUPDATE_HOST"]:
         appini_defines["MOZ_APPUPDATE_HOST"] = CONFIG["MOZ_APPUPDATE_HOST"]
 
