diff --git a/build/variables.py b/build/variables.py
index ef2d6e6800f3c5a09585f52990c827c91c28e652..96606870405616bc12ea9d496e6c29c7563624af 100644
--- a/build/variables.py
+++ b/build/variables.py
@@ -97,7 +97,7 @@ def source_repo_header(output):
         output.write("#define MOZ_SOURCE_STAMP %s\n" % changeset)
 
     if repo and buildconfig.substs.get("MOZ_INCLUDE_SOURCE_INFO"):
-        source = "%s/rev/%s" % (repo, changeset)
+        source = "%s/tree/%s" % (repo, changeset)
         output.write("#define MOZ_SOURCE_REPO %s\n" % repo)
         output.write("#define MOZ_SOURCE_URL %s\n" % source)
 
