diff --git a/build/moz.configure/init.configure b/build/moz.configure/init.configure
index a5b15aa45840193d204f2268d3ea257c4860710c..799364b88cca3a631294266b0dbcb9197915d1db 100644
--- a/build/moz.configure/init.configure
+++ b/build/moz.configure/init.configure
@@ -1197,6 +1197,7 @@ def version_path(path):
 @imports("re")
 def milestone(build_env, build_project, version_path, _):
     versions = []
+    dot_version = ""
     paths = ["config/milestone.txt"]
     if build_project == "js":
         paths = paths * 3
@@ -1224,6 +1225,12 @@ def milestone(build_env, build_project, version_path, _):
 
     milestone, firefox_version, firefox_version_display = versions[:3]
 
+    with open(os.path.join(build_env.topsrcdir, "browser/app/dot/version.txt"), "r") as dv:
+        content = dv.read().splitlines()
+        if not content:
+            die("Could not find a version number in browser/app/dot/version.txt")
+        dot_version = content[-1];
+
     # version.txt content from the project directory if there is one, otherwise
     # the firefox version.
     app_version = versions[3] if len(versions) > 3 else firefox_version
@@ -1273,6 +1280,7 @@ def milestone(build_env, build_project, version_path, _):
         is_early_beta_or_earlier=is_early_beta_or_earlier,
         app_version=app_version,
         app_version_display=app_version_display,
+        dot_version=dot_version,
     )
 
 
@@ -1300,6 +1308,8 @@ set_config("MOZ_APP_VERSION", milestone.app_version)
 set_config("MOZ_APP_VERSION_DISPLAY", milestone.app_version_display)
 add_old_configure_assignment("MOZ_APP_VERSION", milestone.app_version)
 
+set_config("DOT_APP_VERSION", milestone.dot_version)
+add_old_configure_assignment("DOT_APP_VERSION", milestone.dot_version)
 
 # The app update channel is 'default' when not supplied. The value is used in
 # the application's confvars.sh (and is made available to a project specific
