diff --git a/docshell/build/components.conf b/docshell/build/components.conf
index 99a6fd962268b96053c5d3648ce9bff4d2af1c6e..d7e2ae9642c21ea2cd55efbebf05a7f5005d4d92 100644
--- a/docshell/build/components.conf
+++ b/docshell/build/components.conf
@@ -14,6 +14,7 @@ about_pages = [
     'crashcontent',
     'crashparent',
     'credits',
+    'history', # Dot Browser
     'httpsonlyerror',
     'license',
     'logo',
@@ -21,6 +22,7 @@ about_pages = [
     'mozilla',
     'neterror',
     'networking',
+    'newpreferences', # Dot Browser
     'performance',
     'plugins',
     'printpreview',
