diff --git a/dom/base/Document.cpp b/dom/base/Document.cpp
index ce2cbca4b4c5ebb95a0991793131aa53317b862b..21574cfe50adb6c8a8c11cb3fab8d10f7b674801 100644
--- a/dom/base/Document.cpp
+++ b/dom/base/Document.cpp
@@ -17083,8 +17083,9 @@ void Document::RemoveToplevelLoadingDocument(Document* aDoc) {
 
 StylePrefersColorScheme Document::PrefersColorScheme(
     IgnoreRFP aIgnoreRFP) const {
+  NS_WARNING(StaticPrefs::privacy_resistFingerprinting_ignoreColorScheme());
   if (aIgnoreRFP == IgnoreRFP::No &&
-      nsContentUtils::ShouldResistFingerprinting(this)) {
+      StaticPrefs::privacy_resistFingerprinting_ignoreColorScheme()) {
     return StylePrefersColorScheme::Light;
   }
 
