diff --git a/dom/base/Document.cpp b/dom/base/Document.cpp
index 48af2c30f3604a94e6bc1da7848aed3c8ac11c1e..7cb568caae94d0fe03ff02ef8d11711e684a8159 100644
--- a/dom/base/Document.cpp
+++ b/dom/base/Document.cpp
@@ -16821,8 +16821,9 @@ void Document::RemoveToplevelLoadingDocument(Document* aDoc) {
 
 StylePrefersColorScheme Document::PrefersColorScheme(
     IgnoreRFP aIgnoreRFP) const {
+    NS_WARNING(StaticPrefs::privacy_resistFingerprinting_ignoreColorScheme());
   if (aIgnoreRFP == IgnoreRFP::No &&
-      nsContentUtils::ShouldResistFingerprinting(this)) {
+      StaticPrefs::privacy_resistFingerprinting_ignoreColorScheme()) {
     return StylePrefersColorScheme::Light;
   }
 
