diff --git a/dom/base/nsGlobalWindowOuter.cpp b/dom/base/nsGlobalWindowOuter.cpp
index f1b9496cb8c7ebdaec68cec1774cd1b7f92ac88a..79ce368c06eeed08a53d8fc85de94e1fa3291808 100644
--- a/dom/base/nsGlobalWindowOuter.cpp
+++ b/dom/base/nsGlobalWindowOuter.cpp
@@ -4815,7 +4815,7 @@ void nsGlobalWindowOuter::MakeMessageWithPrincipal(
         }
       }
       if (!aUseHostPort || NS_FAILED(rv)) {
-        rv = aSubjectPrincipal->GetExposablePrePath(contentDesc);
+        rv = aSubjectPrincipal->GetAsciiHost(contentDesc);
       }
       if (NS_SUCCEEDED(rv) && !contentDesc.IsEmpty()) {
         NS_ConvertUTF8toUTF16 ucsPrePath(contentDesc);
@@ -4902,6 +4902,9 @@ bool nsGlobalWindowOuter::AlertOrConfirm(bool aAlert, const nsAString& aMessage,
     // Just silently return.  In the case of alert(), the return value is
     // ignored.  In the case of confirm(), returning false is the same thing as
     // would happen if the user cancels.
+
+    uint8_t artificialDelay = rand() % 2000 + 200;
+    PR_Sleep(PR_MillisecondsToInterval(artificialDelay));
     return false;
   }
 
