diff --git a/dom/base/Document.cpp b/dom/base/Document.cpp
index 93baac5ba84922575d8d334533f553f3ad69cfa0..f68baf217852e392bb7a78d62ba4815fe1ca04bf 100644
--- a/dom/base/Document.cpp
+++ b/dom/base/Document.cpp
@@ -16984,8 +16984,9 @@ void Document::RemoveToplevelLoadingDocument(Document* aDoc) {
 
 StylePrefersColorScheme Document::PrefersColorScheme(
     IgnoreRFP aIgnoreRFP) const {
+  NS_WARNING(StaticPrefs::privacy_resistFingerprinting_ignoreColorScheme());
   if (aIgnoreRFP == IgnoreRFP::No &&
-      nsContentUtils::ShouldResistFingerprinting(this)) {
+      StaticPrefs::privacy_resistFingerprinting_ignoreColorScheme()) {
     return StylePrefersColorScheme::Light;
   }
 
