diff --git a/layout/base/PresShell.cpp b/layout/base/PresShell.cpp
index 18b8a562d20cdb6f4278899533b17ffaf3517f5e..896656a8fea2dabb15ae0098b401e432e622fd4d 100644
--- a/layout/base/PresShell.cpp
+++ b/layout/base/PresShell.cpp
@@ -5288,7 +5288,7 @@ nscolor PresShell::GetDefaultBackgroundColorToDraw() {
       doc->PrefersColorScheme(Document::IgnoreRFP::Yes) ==
           StylePrefersColorScheme::Dark) {
     // Use --in-content-page-background for prefers-color-scheme: dark.
-    return NS_RGB(0x2A, 0x2A, 0x2E);
+    return NS_RGB(0x1C, 0x1C, 0x1C);
   }
 
   return backgroundColor;
