diff --git a/layout/base/PresShell.cpp b/layout/base/PresShell.cpp
index fde6c5727b8faed85e58fa658d687322f93d2383..e02632b3fb781a03b3c667969beb9775473a8e8e 100644
--- a/layout/base/PresShell.cpp
+++ b/layout/base/PresShell.cpp
@@ -5293,7 +5293,7 @@ nscolor PresShell::GetDefaultBackgroundColorToDraw() {
       doc->PrefersColorScheme(Document::IgnoreRFP::Yes) ==
           StylePrefersColorScheme::Dark) {
     // Use --in-content-page-background for prefers-color-scheme: dark.
-    return NS_RGB(0x2A, 0x2A, 0x2E);
+    return NS_RGB(0x1C, 0x1C, 0x1C);
   }
 
   return backgroundColor;
