diff --git a/layout/generic/nsImageFrame.cpp b/layout/generic/nsImageFrame.cpp
index 7ab3790f58bd642f26d1d00e2ce2ba3910ab2a0f..79b7d33d89b0368aee9a97a0ad7b561c952db86f 100644
--- a/layout/generic/nsImageFrame.cpp
+++ b/layout/generic/nsImageFrame.cpp
@@ -2547,7 +2547,7 @@ nsresult nsImageFrame::LoadIcons(nsPresContext* aPresContext) {
   NS_ASSERTION(!gIconLoad, "called LoadIcons twice");
 
   constexpr auto loadingSrc = u"resource://gre-resources/loading-image.png"_ns;
-  constexpr auto brokenSrc = u"resource://gre-resources/broken-image.png"_ns;
+  constexpr auto brokenSrc = u"resource://gre-resources/broken-image.svg"_ns;
 
   gIconLoad = new IconLoad();
 
