diff --git a/layout/base/PresShell.cpp b/layout/base/PresShell.cpp
index 05d899547c5c074f114b77ae405675bfc92e4423..e2211e2a6cce5e000735ccf0630bc6fa6424a39e 100644
--- a/layout/base/PresShell.cpp
+++ b/layout/base/PresShell.cpp
@@ -5286,8 +5286,8 @@ nscolor PresShell::GetDefaultBackgroundColorToDraw() {
       doc->PrefersColorScheme(Document::IgnoreRFP::Yes) ==
           StylePrefersColorScheme::Dark) {
     // Use --in-content-page-background for prefers-color-scheme: dark.
-    return StaticPrefs::browser_proton_enabled() ? NS_RGB(0x1C, 0x1B, 0x22)
-                                                 : NS_RGB(0x2A, 0x2A, 0x2E);
+    return StaticPrefs::browser_proton_enabled() ? NS_RGB(0x1C, 0x1C, 0x1C)
+                                                 : NS_RGB(0x1C, 0x1C, 0x1C);
   }
 
   return backgroundColor;
