diff --git a/other-licenses/7zstub/firefox/resource.rc b/other-licenses/7zstub/firefox/resource.rc
index 3a50a450c3fd7f006166537f334b55715c22702a..303596f6f5bb65d69b186124a76c088a020c3190 100644
--- a/other-licenses/7zstub/firefox/resource.rc
+++ b/other-licenses/7zstub/firefox/resource.rc
@@ -16,13 +16,13 @@ BEGIN
   BEGIN
       BLOCK "040904b0"
       BEGIN
-          VALUE "CompanyName", "Mozilla"
-          VALUE "FileDescription", "Firefox"
+          VALUE "CompanyName", "Dot HQ"
+          VALUE "FileDescription", "Dot Browser"
           VALUE "FileVersion", MY_VERSION
           VALUE "InternalName", "7zSD.sfx"
-          VALUE "LegalCopyright", "Mozilla"
+          VALUE "LegalCopyright", "Mozilla and Dot HQ"
           VALUE "OriginalFilename", "7zSD.sfx"
-          VALUE "ProductName", "Firefox"
+          VALUE "ProductName", "Dot Browser"
           VALUE "ProductVersion", MY_VERSION
       END
   END
