diff --git a/python/mozbuild/mozbuild/controller/building.py b/python/mozbuild/mozbuild/controller/building.py
index 2e33b28b0cb1dd5504f78a4412bb441da78e8d9e..3db9f7b50dda21cbbbce5e2b4b33edeff9aac6cb 100644
--- a/python/mozbuild/mozbuild/controller/building.py
+++ b/python/mozbuild/mozbuild/controller/building.py
@@ -1437,7 +1437,7 @@ class BuildDriver(MozbuildObject):
             # if excessive:
             #    print(EXCESSIVE_SWAP_MESSAGE)
 
-            print("To view resource usage of the build, run |mach " "resource-usage|.")
+            #print("To view resource usage of the build, run |mach " "resource-usage|.")
 
         long_build = monitor.elapsed > 1200
 
@@ -1453,7 +1453,7 @@ class BuildDriver(MozbuildObject):
                     "configuring_build_options.html#sccache"
                 )
         else:
-            output.on_line("Your build was successful!")
+            output.on_line("Your build of Dot Browser was successful!")
 
         # Only for full builds because incremental builders likely don't
         # need to be burdened with this.
@@ -1462,13 +1462,13 @@ class BuildDriver(MozbuildObject):
                 # Fennec doesn't have useful output from just building. We should
                 # arguably make the build action useful for Fennec. Another day...
                 if self.substs["MOZ_BUILD_APP"] != "mobile/android":
-                    print("To take your build for a test drive, run: |mach run|")
+                    print("To take your build for a test drive, run |melon run|.")
                 app = self.substs["MOZ_BUILD_APP"]
-                if app in ("browser", "mobile/android"):
-                    print(
-                        "For more information on what to do now, see "
-                        "https://firefox-source-docs.mozilla.org/setup/contributing_code.html"  # noqa
-                    )
+                #if app in ("browser", "mobile/android"):
+                    #print(
+                    #    "For more information on what to do now, see "
+                    #    "https://firefox-source-docs.mozilla.org/setup/contributing_code.html"  # noqa
+                    #)
             except Exception:
                 # Ignore Exceptions in case we can't find config.status (such
                 # as when doing OSX Universal builds)
