diff --git a/services/settings/dumps/main/moz.build b/services/settings/dumps/main/moz.build
index 1d194acb7eb9a08e3fe7ff92534990a075166c2e..66111fbe8a8d86f72d9240d842d18bc760e5ee52 100644
--- a/services/settings/dumps/main/moz.build
+++ b/services/settings/dumps/main/moz.build
@@ -8,11 +8,11 @@ FINAL_TARGET_FILES.defaults.settings.main += [
     "hijack-blocklists.json",
     "language-dictionaries.json",
     "password-recipes.json",
-    "search-config.json",
-    "search-default-override-allowlist.json",
+    # "search-config.json",
+    # "search-default-override-allowlist.json",
     "search-telemetry.json",
     "sites-classification.json",
-    "top-sites.json",
+    # "top-sites.json",
     "url-classifier-skip-urls.json",
     "websites-with-shared-credential-backends.json",
 ]
