diff --git a/toolkit/modules/AppConstants.jsm b/toolkit/modules/AppConstants.jsm
index f181cf3904c888ef7a9ffc6597ddd5ef15e4bee0..cb2376a839fc94719a91c0af3a93fdb551472fe5 100644
--- a/toolkit/modules/AppConstants.jsm
+++ b/toolkit/modules/AppConstants.jsm
@@ -340,6 +340,8 @@ this.AppConstants = Object.freeze({
   MOZ_WIDGET_TOOLKIT: "@MOZ_WIDGET_TOOLKIT@",
   ANDROID_PACKAGE_NAME: "@ANDROID_PACKAGE_NAME@",
 
+  DOT_APP_VERSION: "@DOT_APP_VERSION@",
+
   DEBUG_JS_MODULES: "@DEBUG_JS_MODULES@",
 
   MOZ_BING_API_CLIENTID: "@MOZ_BING_API_CLIENTID@",
