diff --git a/toolkit/mozapps/extensions/content/aboutaddons.js b/toolkit/mozapps/extensions/content/aboutaddons.js
index abab57dcb4c68113083238ff130b4bbc6d29ba91..7a5cd6912932a93d0bdff18b59dc23716396aee6 100644
--- a/toolkit/mozapps/extensions/content/aboutaddons.js
+++ b/toolkit/mozapps/extensions/content/aboutaddons.js
@@ -86,15 +86,15 @@ const EXTENSION_ICON_URL =
   "chrome://mozapps/skin/extensions/extensionGeneric.svg";
 const BUILTIN_THEME_PREVIEWS = new Map([
   [
-    "default-theme@mozilla.org",
+    "dynamic@themes.dothq.co",
     "chrome://mozapps/content/extensions/default-theme.svg",
   ],
   [
-    "firefox-compact-light@mozilla.org",
+    "light@themes.dothq.co",
     "chrome://mozapps/content/extensions/firefox-compact-light.svg",
   ],
   [
-    "firefox-compact-dark@mozilla.org",
+    "dark@themes.dothq.co",
     "chrome://mozapps/content/extensions/firefox-compact-dark.svg",
   ],
   [
