diff --git a/toolkit/xre/nsAppRunner.cpp b/toolkit/xre/nsAppRunner.cpp
index 27a81620e3ec56e9977be79b7af68ada331f10fb..556d84e60a2e8b1a5a390ee3fb6fadd3c2bcbdf3 100644
--- a/toolkit/xre/nsAppRunner.cpp
+++ b/toolkit/xre/nsAppRunner.cpp
@@ -1980,11 +1980,11 @@ static void ReflectSkeletonUIPrefToRegistry(const char* aPref, void* aData) {
   if (shouldBeEnabled && Preferences::HasUserValue(kPrefThemeId)) {
     nsCString themeId;
     Preferences::GetCString(kPrefThemeId, themeId);
-    if (themeId.EqualsLiteral("default-theme@mozilla.org")) {
+    if (themeId.EqualsLiteral("dynamic@themes.dothq.co")) {
       Unused << SetPreXULSkeletonUIThemeId(ThemeMode::Default);
-    } else if (themeId.EqualsLiteral("firefox-compact-dark@mozilla.org")) {
+    } else if (themeId.EqualsLiteral("dark@themes.dothq.co")) {
       Unused << SetPreXULSkeletonUIThemeId(ThemeMode::Dark);
-    } else if (themeId.EqualsLiteral("firefox-compact-light@mozilla.org")) {
+    } else if (themeId.EqualsLiteral("light@themes.dothq.co")) {
       Unused << SetPreXULSkeletonUIThemeId(ThemeMode::Light);
     } else {
       shouldBeEnabled = false;
