diff --git a/toolkit/modules/Troubleshoot.jsm b/toolkit/modules/Troubleshoot.jsm
index db96f52a3b24a715670126553b42a9e223c0d320..2515686e8f7c9317453a3373a499943cb28eb02c 100644
--- a/toolkit/modules/Troubleshoot.jsm
+++ b/toolkit/modules/Troubleshoot.jsm
@@ -203,6 +203,7 @@ var dataProviders = {
         " " +
         Services.sysinfo.getProperty("build"),
       version: AppConstants.MOZ_APP_VERSION_DISPLAY,
+      dotVersion: AppConstants.DOT_APP_VERSION,
       buildID: Services.appinfo.appBuildID,
       distributionID: Services.prefs
         .getDefaultBranch("")
