diff --git a/toolkit/mozapps/extensions/content/aboutaddons.js b/toolkit/mozapps/extensions/content/aboutaddons.js
index 2beb23784fd9bd4a0d0a9c90c575ad57e470e46f..3abe4d623c172c81e3272a72f61c259525ed035c 100644
--- a/toolkit/mozapps/extensions/content/aboutaddons.js
+++ b/toolkit/mozapps/extensions/content/aboutaddons.js
@@ -86,15 +86,15 @@ const EXTENSION_ICON_URL =
   "chrome://mozapps/skin/extensions/extensionGeneric.svg";
 const BUILTIN_THEME_PREVIEWS = new Map([
   [
-    "default-theme@mozilla.org",
+    "dynamic@themes.dothq.co",
     "chrome://mozapps/content/extensions/default-theme.svg",
   ],
   [
-    "firefox-compact-light@mozilla.org",
+    "light@themes.dothq.co",
     "chrome://mozapps/content/extensions/firefox-compact-light.svg",
   ],
   [
-    "firefox-compact-dark@mozilla.org",
+    "dark@themes.dothq.co",
     "chrome://mozapps/content/extensions/firefox-compact-dark.svg",
   ],
   [
