diff --git a/toolkit/mozapps/extensions/content/aboutaddons.js b/toolkit/mozapps/extensions/content/aboutaddons.js
index 3fccbceab5f9ef98de40277be3739e53de061631..80f51fccba28399a1f427b822d102d292f1e374d 100644
--- a/toolkit/mozapps/extensions/content/aboutaddons.js
+++ b/toolkit/mozapps/extensions/content/aboutaddons.js
@@ -87,15 +87,15 @@ const EXTENSION_ICON_URL =
   "chrome://mozapps/skin/extensions/extensionGeneric.svg";
 const BUILTIN_THEME_PREVIEWS = new Map([
   [
-    "default-theme@mozilla.org",
+    "dynamic@themes.dothq.co",
     "chrome://mozapps/content/extensions/default-theme.svg",
   ],
   [
-    "firefox-compact-light@mozilla.org",
+    "light@themes.dothq.co",
     "chrome://mozapps/content/extensions/firefox-compact-light.svg",
   ],
   [
-    "firefox-compact-dark@mozilla.org",
+    "dark@themes.dothq.co",
     "chrome://mozapps/content/extensions/firefox-compact-dark.svg",
   ],
   [
