diff --git a/toolkit/mozapps/extensions/internal/XPIProvider.jsm b/toolkit/mozapps/extensions/internal/XPIProvider.jsm
index 06773a773c93baa3e5299b74dc9373095447ecd1..c83f46c2bc16e24ae82323c4cb961518416410da 100644
--- a/toolkit/mozapps/extensions/internal/XPIProvider.jsm
+++ b/toolkit/mozapps/extensions/internal/XPIProvider.jsm
@@ -2472,7 +2472,7 @@ var XPIProvider = {
       AddonManagerPrivate.markProviderSafe(this);
 
       this.maybeInstallBuiltinAddon(
-        "default-theme@mozilla.org",
+        "dynamic@themes.dothq.co",
         "1.1",
         "resource://default-theme/"
       );
