diff --git a/toolkit/xre/nsAppRunner.cpp b/toolkit/xre/nsAppRunner.cpp
index 3a9928cee4525652b99856a460eeff9b30a1cffc..57e588039358f3092e8f643b2242a09145584199 100644
--- a/toolkit/xre/nsAppRunner.cpp
+++ b/toolkit/xre/nsAppRunner.cpp
@@ -1979,11 +1979,11 @@ static void ReflectSkeletonUIPrefToRegistry(const char* aPref, void* aData) {
   if (shouldBeEnabled && Preferences::HasUserValue(kPrefThemeId)) {
     nsCString themeId;
     Preferences::GetCString(kPrefThemeId, themeId);
-    if (themeId.EqualsLiteral("default-theme@mozilla.org")) {
+    if (themeId.EqualsLiteral("dynamic@themes.dothq.co")) {
       Unused << SetPreXULSkeletonUIThemeId(ThemeMode::Default);
-    } else if (themeId.EqualsLiteral("firefox-compact-dark@mozilla.org")) {
+    } else if (themeId.EqualsLiteral("dark@themes.dothq.co")) {
       Unused << SetPreXULSkeletonUIThemeId(ThemeMode::Dark);
-    } else if (themeId.EqualsLiteral("firefox-compact-light@mozilla.org")) {
+    } else if (themeId.EqualsLiteral("light@themes.dothq.co")) {
       Unused << SetPreXULSkeletonUIThemeId(ThemeMode::Light);
     } else {
       shouldBeEnabled = false;
