diff --git a/toolkit/xre/nsAppRunner.cpp b/toolkit/xre/nsAppRunner.cpp
index 5eb43aa0ae6320ef1ab47ddf03b4d8fbd4107d86..64980379427876a77b36a4cc3c383cabc9654e19 100644
--- a/toolkit/xre/nsAppRunner.cpp
+++ b/toolkit/xre/nsAppRunner.cpp
@@ -1991,11 +1991,11 @@ static void ReflectSkeletonUIPrefToRegistry(const char* aPref, void* aData) {
   if (shouldBeEnabled && Preferences::HasUserValue(kPrefThemeId)) {
     nsCString themeId;
     Preferences::GetCString(kPrefThemeId, themeId);
-    if (themeId.EqualsLiteral("default-theme@mozilla.org")) {
+    if (themeId.EqualsLiteral("dynamic@themes.dothq.co")) {
       Unused << SetPreXULSkeletonUIThemeId(ThemeMode::Default);
-    } else if (themeId.EqualsLiteral("firefox-compact-dark@mozilla.org")) {
+    } else if (themeId.EqualsLiteral("dark@themes.dothq.co")) {
       Unused << SetPreXULSkeletonUIThemeId(ThemeMode::Dark);
-    } else if (themeId.EqualsLiteral("firefox-compact-light@mozilla.org")) {
+    } else if (themeId.EqualsLiteral("light@themes.dothq.co")) {
       Unused << SetPreXULSkeletonUIThemeId(ThemeMode::Light);
     } else {
       shouldBeEnabled = false;
