diff --git a/toolkit/modules/LightweightThemeConsumer.jsm b/toolkit/modules/LightweightThemeConsumer.jsm
index 10a2cd67fb045a4d6fe9e327da455c1b4cf7814f..94473fad0829c13fdd18f4b47d198532add735b3 100644
--- a/toolkit/modules/LightweightThemeConsumer.jsm
+++ b/toolkit/modules/LightweightThemeConsumer.jsm
@@ -6,9 +6,9 @@ var EXPORTED_SYMBOLS = ["LightweightThemeConsumer"];
 
 const { Services } = ChromeUtils.import("resource://gre/modules/Services.jsm");
 
-const DEFAULT_THEME_ID = "default-theme@mozilla.org";
-const LIGHT_THEME_ID = "firefox-compact-light@mozilla.org";
-const DARK_THEME_ID = "firefox-compact-dark@mozilla.org";
+const DEFAULT_THEME_ID = "dynamic@themes.dothq.co";
+const LIGHT_THEME_ID = "light@themes.dothq.co";
+const DARK_THEME_ID = "dark@themes.dothq.co";
 
 ChromeUtils.defineModuleGetter(
   this,
@@ -272,7 +272,7 @@ LightweightThemeConsumer.prototype = {
     const useDarkTheme =
       themeData.darkTheme &&
       this.darkThemeMediaQuery?.matches &&
-      (themeData.darkTheme.id != DEFAULT_THEME_ID ||
+      (themeData.darkTheme.id != 'ThisThemeWillNotHaveTheDarkThemeKey,useADifferentID' ||
         AppConstants.platform != "linux");
 
     let theme = useDarkTheme ? themeData.darkTheme : themeData.theme;
