diff --git a/toolkit/modules/SubDialog.jsm b/toolkit/modules/SubDialog.jsm
index 68d773afc88714fbc21c01667bbf7a2a4194bcd9..ca9a314b91f363707f5a10ee1bf8db0b7f575d4c 100644
--- a/toolkit/modules/SubDialog.jsm
+++ b/toolkit/modules/SubDialog.jsm
@@ -172,9 +172,8 @@ SubDialog.prototype = {
 
     // If the parent is chrome we also need open the dialog as chrome, otherwise
     // the openDialog call will fail.
-    let dialogFeatures = `resizable,dialog=no,centerscreen,chrome=${
-      this._window?.isChromeWindow ? "yes" : "no"
-    }`;
+    let dialogFeatures = `resizable,dialog=no,centerscreen,chrome=${this._window?.isChromeWindow ? "yes" : "no"
+      }`;
     if (features) {
       dialogFeatures = `${features},${dialogFeatures}`;
     }
@@ -195,8 +194,8 @@ SubDialog.prototype = {
     this._box.setAttribute(
       "resizable",
       featureParams.has("resizable") &&
-        featureParams.get("resizable") != "no" &&
-        featureParams.get("resizable") != "0"
+      featureParams.get("resizable") != "no" &&
+      featureParams.get("resizable") != "0"
     );
   },
 
@@ -470,6 +469,8 @@ SubDialog.prototype = {
     if (this._window.isChromeWindow) {
       boxMinWidth = `min(80vw, ${boxMinWidth})`;
     }
+
+    if (docEl.getAttribute("autowidth") == "true") boxMinWidth = "auto";
     this._box.style.minWidth = boxMinWidth;
 
     this.resizeVertically();
@@ -566,13 +567,13 @@ SubDialog.prototype = {
     } else {
       Cu.reportError(
         "This dialog (" +
-          this._frame.contentWindow.location.href +
-          ") " +
-          "set a height in non-px-non-em units ('" +
-          frameHeight +
-          "'), " +
-          "which is likely to lead to bad sizing in in-content preferences. " +
-          "Please consider changing this."
+        this._frame.contentWindow.location.href +
+        ") " +
+        "set a height in non-px-non-em units ('" +
+        frameHeight +
+        "'), " +
+        "which is likely to lead to bad sizing in in-content preferences. " +
+        "Please consider changing this."
       );
       comparisonFrameHeight = parseFloat(frameHeight);
     }
