diff --git a/toolkit/mozapps/extensions/internal/XPIProvider.jsm b/toolkit/mozapps/extensions/internal/XPIProvider.jsm
index 0599e221ed1eac4b894ae9aeab3b4f958c4a48bd..b53d188014ca3ce63ee0439937ebe584cb07a7e0 100644
--- a/toolkit/mozapps/extensions/internal/XPIProvider.jsm
+++ b/toolkit/mozapps/extensions/internal/XPIProvider.jsm
@@ -2472,7 +2472,7 @@ var XPIProvider = {
       AddonManagerPrivate.markProviderSafe(this);
 
       this.maybeInstallBuiltinAddon(
-        "default-theme@mozilla.org",
+        "dynamic@themes.dothq.co",
         "1.2",
         "resource://default-theme/"
       );
