diff --git a/toolkit/modules/moz.build b/toolkit/modules/moz.build
index 3626b9f83402814fe27d223df07d8b96e9ae14e6..b83e5298e5eaed73903fa45ae2745cb5edaf0d29 100644
--- a/toolkit/modules/moz.build
+++ b/toolkit/modules/moz.build
@@ -276,6 +276,7 @@ if CONFIG["MOZ_WIDGET_TOOLKIT"] == "windows":
 
 for var in (
     "ANDROID_PACKAGE_NAME",
+    "DOT_APP_VERSION",
     "MOZ_APP_NAME",
     "MOZ_APP_BASENAME",
     "MOZ_APP_DISPLAYNAME",
