diff --git a/toolkit/components/moz.build b/toolkit/components/moz.build
index dbcab1465abfdcfab83587e373851d64357fea04..dff7f2c61e5ddb71e3a88771bb97f26286d07856 100644
--- a/toolkit/components/moz.build
+++ b/toolkit/components/moz.build
@@ -33,6 +33,7 @@ DIRS += [
     "contextualidentity",
     "crashes",
     "crashmonitor",
+    "dotprotocol",
     "downloads",
     "enterprisepolicies",
     "extensions",
