diff --git a/toolkit/modules/Troubleshoot.jsm b/toolkit/modules/Troubleshoot.jsm
index fa61c1eb877577624dbc07c690a576ee810c602f..43c8e09ce1464db1e88a05c664bd04af7f3fa0a7 100644
--- a/toolkit/modules/Troubleshoot.jsm
+++ b/toolkit/modules/Troubleshoot.jsm
@@ -204,6 +204,7 @@ var dataProviders = {
         " " +
         Services.sysinfo.getProperty("build"),
       version: AppConstants.MOZ_APP_VERSION_DISPLAY,
+      dotVersion: AppConstants.DOT_APP_VERSION,
       buildID: Services.appinfo.appBuildID,
       distributionID: Services.prefs
         .getDefaultBranch("")
