diff --git a/toolkit/mozapps/extensions/internal/XPIDatabase.jsm b/toolkit/mozapps/extensions/internal/XPIDatabase.jsm
index de19b25893ad79a7e1e187906e0a0c02045df104..7f9c9d86be939b2ff04c2ac5b7d3bfcd51f1eb76 100644
--- a/toolkit/mozapps/extensions/internal/XPIDatabase.jsm
+++ b/toolkit/mozapps/extensions/internal/XPIDatabase.jsm
@@ -95,7 +95,7 @@ const KEY_APP_GLOBAL = "app-global";
 const KEY_APP_PROFILE = "app-profile";
 const KEY_APP_TEMPORARY = "app-temporary";
 
-const DEFAULT_THEME_ID = "default-theme@mozilla.org";
+const DEFAULT_THEME_ID = "dynamic@themes.dothq.co";
 
 // Properties to cache and reload when an addon installation is pending
 const PENDING_INSTALL_METADATA = [
