diff --git a/toolkit/components/moz.build b/toolkit/components/moz.build
index 474e9561ec803723ea79da2c2e1accab5a81d578..5cb2ea1ff4ca5edd6a76356eb88c4bd52fc65d6c 100644
--- a/toolkit/components/moz.build
+++ b/toolkit/components/moz.build
@@ -33,6 +33,7 @@ DIRS += [
     "contextualidentity",
     "crashes",
     "crashmonitor",
+    "dotprotocol",
     "downloads",
     "enterprisepolicies",
     "extensions",
