diff --git a/toolkit/content/aboutSupport.js b/toolkit/content/aboutSupport.js
index 72ea17475ac6a781d15693063f67b4640d0900c9..06458b05588e61ee0199e23ab62be9621a5ec073 100644
--- a/toolkit/content/aboutSupport.js
+++ b/toolkit/content/aboutSupport.js
@@ -95,7 +95,7 @@ var snapshotFormatters = {
       Ci.nsIFile
     ).path;
     $("supportLink").href = data.supportURL;
-    let version = AppConstants.MOZ_APP_VERSION_DISPLAY;
+    let version = AppConstants.DOT_APP_VERSION;
     if (data.vendor) {
       version += " (" + data.vendor + ")";
     }
