diff --git a/toolkit/mozapps/extensions/internal/XPIProvider.jsm b/toolkit/mozapps/extensions/internal/XPIProvider.jsm
index ac5bdacb6de917005dd568b676bec7238a23e176..e1a49536477125f5a507814d16f7a57f712193a9 100644
--- a/toolkit/mozapps/extensions/internal/XPIProvider.jsm
+++ b/toolkit/mozapps/extensions/internal/XPIProvider.jsm
@@ -2469,7 +2469,7 @@ var XPIProvider = {
       AddonManagerPrivate.markProviderSafe(this);
 
       this.maybeInstallBuiltinAddon(
-        "default-theme@mozilla.org",
+        "dynamic@themes.dothq.co",
         "1.1",
         "resource://default-theme/"
       );
