diff --git a/toolkit/components/search/SearchEngineSelector.jsm b/toolkit/components/search/SearchEngineSelector.jsm
index 9f614771820090f066b6191a3349d7b5add670cd..48aeb8e0d81f558f32dc068401ae62e04bc7c422 100644
--- a/toolkit/components/search/SearchEngineSelector.jsm
+++ b/toolkit/components/search/SearchEngineSelector.jsm
@@ -196,6 +196,77 @@ class SearchEngineSelector {
   }) {
     if (!this._configuration) {
       await this.getEngineConfiguration();
+
+      if(!this._configuration) {
+        this._configuration = [
+          {
+              "webExtension": {
+                  "id": "ddg@search.mozilla.org"
+              },
+              "appliesTo": [
+                  {
+                      "default": "yes-if-no-other",
+                      "included": {
+                          "everywhere": true
+                      }
+                  }
+              ],
+              "extraParams": []
+          },
+          {
+              "webExtension": {
+                  "id": "ecosia@search.mozilla.org"
+              },
+              "appliesTo": [
+                  {
+                      "included": {
+                          "everywhere": true
+                      }
+                  }
+              ],
+              "extraParams": []
+          },
+          {
+              "webExtension": {
+                  "id": "google@search.mozilla.org"
+              },
+              "appliesTo": [
+                  {
+                      "included": {
+                          "everywhere": true
+                      }
+                  }
+              ],
+              "extraParams": []
+          },
+          {
+              "webExtension": {
+                  "id": "qwant@search.mozilla.org"
+              },
+              "appliesTo": [
+                  {
+                      "included": {
+                          "everywhere": true
+                      }
+                  }
+              ],
+              "extraParams": []
+          },
+          {
+              "webExtension": {
+                  "id": "bing@search.mozilla.org"
+              },
+              "appliesTo": [
+                  {
+                      "included": {
+                          "everywhere": true
+                      }
+                  }
+              ],
+              "extraParams": []
+          }
+        ]
+      }
     }
     let name = getAppInfo("name");
     let version = getAppInfo("version");
