diff --git a/browser/components/BrowserGlue.jsm b/browser/components/BrowserGlue.jsm
index 858e24b8f91089545c84847f66714fea033b4b48..69e06daebd084ed9f5e7b674b99dc6c43de09af5 100644
--- a/browser/components/BrowserGlue.jsm
+++ b/browser/components/BrowserGlue.jsm
@@ -7,6 +7,7 @@ var EXPORTED_SYMBOLS = [
   "BrowserGlue",
   "ContentPermissionPrompt",
   "DefaultBrowserCheck",
+  "JSWINDOWACTORS"
 ];
 
 const { XPCOMUtils } = ChromeUtils.import(
@@ -166,7 +167,7 @@ let JSPROCESSACTORS = {
               "RefreshBlocker",
               "all"
             );
-          } catch (ex) {}
+          } catch (ex) { }
         }
       });
     },
@@ -770,8 +771,8 @@ let JSWINDOWACTORS = {
     !Services.startup.showedPreXULSkeletonUI &&
     Services.prefs.getCharPref(
       "extensions.activeThemeID",
-      "default-theme@mozilla.org"
-    ) != "default-theme@mozilla.org"
+      Services.builtInThemes.DEFAULT_THEME_ID
+    ) != Services.builtInThemes.DEFAULT_THEME_ID
   ) {
     return;
   }
@@ -860,19 +861,19 @@ if (AppConstants.MOZ_CRASHREPORTER) {
   });
 }
 
-XPCOMUtils.defineLazyGetter(this, "gBrandBundle", function() {
+XPCOMUtils.defineLazyGetter(this, "gBrandBundle", function () {
   return Services.strings.createBundle(
     "chrome://branding/locale/brand.properties"
   );
 });
 
-XPCOMUtils.defineLazyGetter(this, "gBrowserBundle", function() {
+XPCOMUtils.defineLazyGetter(this, "gBrowserBundle", function () {
   return Services.strings.createBundle(
     "chrome://browser/locale/browser.properties"
   );
 });
 
-XPCOMUtils.defineLazyGetter(this, "gTabbrowserBundle", function() {
+XPCOMUtils.defineLazyGetter(this, "gTabbrowserBundle", function () {
   return Services.strings.createBundle(
     "chrome://browser/locale/tabbrowser.properties"
   );
@@ -936,7 +937,7 @@ function BrowserGlue() {
     "nsIUserIdleService"
   );
 
-  XPCOMUtils.defineLazyGetter(this, "_distributionCustomizer", function() {
+  XPCOMUtils.defineLazyGetter(this, "_distributionCustomizer", function () {
     const { DistributionCustomizer } = ChromeUtils.import(
       "resource:///modules/distribution.js"
     );
@@ -1238,7 +1239,8 @@ BrowserGlue.prototype = {
     os.addObserver(this, "handlersvc-store-initialized");
 
     ActorManagerParent.addJSProcessActors(JSPROCESSACTORS);
-    ActorManagerParent.addJSWindowActors(JSWINDOWACTORS);
+    /* This is now handled in dot/modules/glue.ts */
+    // ActorManagerParent.addJSWindowActors(JSWINDOWACTORS);
 
     this._flashHangCount = 0;
     this._firstWindowReady = new Promise(
@@ -1366,22 +1368,22 @@ BrowserGlue.prototype = {
 
     SessionStore.init();
 
-    AddonManager.maybeInstallBuiltinAddon(
-      "firefox-compact-light@mozilla.org",
-      "1.2",
-      "resource://builtin-themes/light/"
-    );
-    AddonManager.maybeInstallBuiltinAddon(
-      "firefox-compact-dark@mozilla.org",
-      "1.2",
-      "resource://builtin-themes/dark/"
-    );
-
-    AddonManager.maybeInstallBuiltinAddon(
-      "firefox-alpenglow@mozilla.org",
-      "1.4",
-      "resource://builtin-themes/alpenglow/"
-    );
+    // AddonManager.maybeInstallBuiltinAddon(
+    //   "firefox-compact-light@mozilla.org",
+    //   "1.2",
+    //   "resource://builtin-themes/light/"
+    // );
+    // AddonManager.maybeInstallBuiltinAddon(
+    //   "firefox-compact-dark@mozilla.org",
+    //   "1.2",
+    //   "resource://builtin-themes/dark/"
+    // );
+
+    // AddonManager.maybeInstallBuiltinAddon(
+    //   "firefox-alpenglow@mozilla.org",
+    //   "1.4",
+    //   "resource://builtin-themes/alpenglow/"
+    // );
 
     if (AppConstants.MOZ_NORMANDY) {
       Normandy.init();
@@ -1575,7 +1577,7 @@ BrowserGlue.prototype = {
     let scaling = aWindow.devicePixelRatio * 100;
     try {
       Services.telemetry.getHistogramById("DISPLAY_SCALING").add(scaling);
-    } catch (ex) {}
+    } catch (ex) { }
   },
 
   _collectStartupConditionsTelemetry() {
@@ -1659,7 +1661,7 @@ BrowserGlue.prototype = {
           "resource://gre/modules/UpdateUtils.jsm",
           {}
         ).UpdateUtils.UpdateChannel;
-      } catch (ex) {}
+      } catch (ex) { }
       if (updateChannel) {
         let uninstalledValue = WindowsRegistry.readRegKey(
           Ci.nsIWindowsRegKey.ROOT_KEY_CURRENT_USER,
@@ -1815,7 +1817,7 @@ BrowserGlue.prototype = {
     );
     let categoryPref;
     switch (
-      Services.prefs.getStringPref("browser.contentblocking.category", null)
+    Services.prefs.getStringPref("browser.contentblocking.category", null)
     ) {
       case "standard":
         categoryPref = 0;
@@ -2049,7 +2051,7 @@ BrowserGlue.prototype = {
           }
         });
       },
-      onCloseWindow() {},
+      onCloseWindow() { },
     };
 
     Services.prefs.addObserver(PREF_ION_ID, _checkIonPref);
@@ -2088,7 +2090,7 @@ BrowserGlue.prototype = {
           }
         });
       },
-      onCloseWindow() {},
+      onCloseWindow() { },
     };
 
     // Update all open windows if the pref changes.
@@ -2396,7 +2398,7 @@ BrowserGlue.prototype = {
           let { setTimeout } = ChromeUtils.import(
             "resource://gre/modules/Timer.jsm"
           );
-          setTimeout(function() {
+          setTimeout(function () {
             Services.tm.idleDispatchToMainThread(
               Services.startup.trackStartupCrashEnd
             );
@@ -2622,7 +2624,7 @@ BrowserGlue.prototype = {
         this._gmpInstallManager = new obj.GMPInstallManager();
         // We don't really care about the results, if someone is interested they
         // can check the log.
-        this._gmpInstallManager.simpleCheckAndInstall().catch(() => {});
+        this._gmpInstallManager.simpleCheckAndInstall().catch(() => { });
       },
 
       () => {
@@ -2687,7 +2689,7 @@ BrowserGlue.prototype = {
 
   _addBreachAlertsPrefObserver() {
     const BREACH_ALERTS_PREF = "signon.management.page.breach-alerts.enabled";
-    const clearVulnerablePasswordsIfBreachAlertsDisabled = async function() {
+    const clearVulnerablePasswordsIfBreachAlertsDisabled = async function () {
       if (!Services.prefs.getBoolPref(BREACH_ALERTS_PREF)) {
         await LoginBreaches.clearAllPotentiallyVulnerablePasswords();
       }
@@ -2886,7 +2888,7 @@ BrowserGlue.prototype = {
       if (importBookmarksHTML) {
         importBookmarks = true;
       }
-    } catch (ex) {}
+    } catch (ex) { }
 
     // Support legacy bookmarks.html format for apps that depend on that format.
     let autoExportHTML = Services.prefs.getBoolPref(
@@ -2915,7 +2917,7 @@ BrowserGlue.prototype = {
           await this._backupBookmarks();
           importBookmarks = true;
         }
-      } catch (ex) {}
+      } catch (ex) { }
 
       // This may be reused later, check for "=== undefined" to see if it has
       // been populated already.
@@ -3074,14 +3076,14 @@ BrowserGlue.prototype = {
    * If a backup for today doesn't exist, this creates one.
    */
   _backupBookmarks: function BG__backupBookmarks() {
-    return (async function() {
+    return (async function () {
       let lastBackupFile = await PlacesBackups.getMostRecentBackup();
       // Should backup bookmarks if there are no backups or the maximum
       // interval between backups elapsed.
       if (
         !lastBackupFile ||
         new Date() - PlacesBackups.getDateForFile(lastBackupFile) >
-          BOOKMARKS_BACKUP_MIN_INTERVAL_DAYS * 86400000
+        BOOKMARKS_BACKUP_MIN_INTERVAL_DAYS * 86400000
       ) {
         let maxBackups = Services.prefs.getIntPref(
           "browser.bookmarks.max_backups"
@@ -3186,7 +3188,7 @@ BrowserGlue.prototype = {
     if (
       currentUIVersion < 65 &&
       Services.prefs.getCharPref("general.config.filename", "") ==
-        "dsengine.cfg"
+      "dsengine.cfg"
     ) {
       let searchInitializedPromise = new Promise(resolve => {
         if (Services.search.isInitialized) {
@@ -3402,7 +3404,7 @@ BrowserGlue.prototype = {
         !Services.prefs.getBoolPref("media.autoplay.allow-muted") &&
         !Services.prefs.prefHasUserValue("media.autoplay.default") &&
         Services.prefs.getIntPref("media.autoplay.default") ==
-          Ci.nsIAutoplay.BLOCKED
+        Ci.nsIAutoplay.BLOCKED
       ) {
         Services.prefs.setIntPref(
           "media.autoplay.default",
@@ -3548,8 +3550,8 @@ BrowserGlue.prototype = {
       } catch (error) {
         Cu.reportError(
           "Could not access the AddonManager to upgrade the profile. This is most " +
-            "likely because the upgrader is being run from an xpcshell test where " +
-            "the AddonManager is not initialized."
+          "likely because the upgrader is being run from an xpcshell test where " +
+          "the AddonManager is not initialized."
         );
       }
       Promise.resolve(addonPromise).then(addon => {
@@ -4498,7 +4500,7 @@ const ContentPermissionIntegration = {
   },
 };
 
-function ContentPermissionPrompt() {}
+function ContentPermissionPrompt() { }
 
 ContentPermissionPrompt.prototype = {
   classID: Components.ID("{d8903bf6-68d5-4e97-bcd1-e4d3012f721a}"),
@@ -4816,7 +4818,7 @@ var JawsScreenReaderVersionCheck = {
     if (!win || !win.gBrowser || !win.gBrowser.selectedBrowser) {
       Services.console.logStringMessage(
         "Content access support for older versions of JAWS is disabled " +
-          "due to compatibility issues with this version of Firefox."
+        "due to compatibility issues with this version of Firefox."
       );
       this._prompted = false;
       return;
@@ -5412,7 +5414,7 @@ var AboutHomeStartupCache = {
 
         this.log.trace(
           "Writing the page data is complete. Now opening the " +
-            "script output stream."
+          "script output stream."
         );
 
         let scriptOutputStream;
@@ -5557,7 +5559,7 @@ var AboutHomeStartupCache = {
       if (this._cacheDeferred) {
         this.log.error(
           "A privileged about content process shut down while cache streams " +
-            "were still en route."
+          "were still en route."
         );
         // The crash occurred while we were waiting on cache input streams to
         // be returned to us. Resolve with null streams instead.
@@ -5655,7 +5657,7 @@ var AboutHomeStartupCache = {
       ) {
         this.log.error(
           "Somehow got a success result despite having never " +
-            "successfully sent down the cache streams"
+          "successfully sent down the cache streams"
         );
         this.recordResult(this._cacheDeferredResultScalar);
       } else {
