diff --git a/devtools/client/framework/tab-descriptor-factory.js b/devtools/client/framework/tab-descriptor-factory.js
index 92a3781528a1835a21c1b75531651b615b94fb61..bc6e6c72b7f8dce41c94748cfafb9a3ad97e2987 100644
--- a/devtools/client/framework/tab-descriptor-factory.js
+++ b/devtools/client/framework/tab-descriptor-factory.js
@@ -107,7 +107,7 @@ exports.TabDescriptorFactory = {
    * descriptor for a tab without creating a descriptor.
    * @return true/false
    */
-  isKnownTab: function(tab) {
+  isKnownTab: function (tab) {
     return descriptors.has(tab);
   },
 };
