diff --git a/browser/app.mozbuild b/browser/app.mozbuild
index 454933eb514997d9dbf99a976a250d290a9a6199..4f83298956d0b49df916d6ee40f3419b987a1830 100644
--- a/browser/app.mozbuild
+++ b/browser/app.mozbuild
@@ -3,6 +3,7 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
+include('/dot/moz.build')
 include('/toolkit/toolkit.mozbuild')
 
 if CONFIG['MOZ_EXTENSIONS']:
