diff --git a/browser/confvars.sh b/browser/confvars.sh
index 92871c9516f98e065c5240a4a1cc7ead1de8ef9d..e8b399bed98b8fd7b5aeb722641cd0a19307aa7b 100755
--- a/browser/confvars.sh
+++ b/browser/confvars.sh
@@ -3,7 +3,7 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
-MOZ_APP_VENDOR=Mozilla
+MOZ_APP_VENDOR="Dot HQ"
 MOZ_UPDATER=1
 
 if test "$OS_ARCH" = "WINNT"; then
@@ -26,7 +26,7 @@ if test "$OS_ARCH" = "WINNT"; then
   fi
 fi
 
-BROWSER_CHROME_URL=chrome://browser/content/browser.xhtml
+BROWSER_CHROME_URL=chrome://dot/content/browser.html
 
 # MOZ_APP_DISPLAYNAME will be set by branding/configure.sh
 # MOZ_BRANDING_DIRECTORY is the default branding directory used when none is
@@ -34,8 +34,8 @@ BROWSER_CHROME_URL=chrome://browser/content/browser.xhtml
 # For mozilla-beta, mozilla-release, or mozilla-central repositories, use
 # "unofficial" branding.
 # For the mozilla-aurora repository, use "aurora".
-MOZ_BRANDING_DIRECTORY=browser/branding/unofficial
-MOZ_OFFICIAL_BRANDING_DIRECTORY=browser/branding/official
+MOZ_BRANDING_DIRECTORY=browser/branding/dot
+MOZ_OFFICIAL_BRANDING_DIRECTORY=browser/branding/dot
 MOZ_APP_ID={ec8030f7-c20a-464f-9b0e-13a3a9e97384}
 
 MOZ_PROFILE_MIGRATOR=1
