diff --git a/docshell/build/components.conf b/docshell/build/components.conf
index 99a6fd962268b96053c5d3648ce9bff4d2af1c6e..22828e06e553539ad5453738329d347b9711cc2a 100644
--- a/docshell/build/components.conf
+++ b/docshell/build/components.conf
@@ -10,7 +10,7 @@ about_pages = [
     'buildconfig',
     'certificate',
     'checkerboard',
-    'config',
+    # 'config', // moved to dot/about/components.conf
     'crashcontent',
     'crashparent',
     'credits',
