diff --git a/config/baseconfig.mk b/config/baseconfig.mk
index 8f3e920516c7a484cfd23f3067bbc31d5fc3712f..c6179844ab115c60c656afbb7fc51c21f432a9af 100644
--- a/config/baseconfig.mk
+++ b/config/baseconfig.mk
@@ -43,7 +43,7 @@ ifeq ($(MOZ_BUILD_APP),tools/rusttests)
 ALL_TIERS := pre-export export rusttests
 else
 # All possible tiers
-ALL_TIERS := artifact win32-artifact android-fat-aar-artifact pre-export export pre-compile rust compile misc libs android-stage-package android-archive-geckoview tools check
+ALL_TIERS := artifact win32-artifact android-fat-aar-artifact pre-export export pre-compile rust frontend compile misc libs android-stage-package android-archive-geckoview tools check
 endif
 
 # All tiers that may be used manually via `mach build $tier`
