diff --git a/toolkit/modules/LightweightThemeConsumer.jsm b/toolkit/modules/LightweightThemeConsumer.jsm
index 10a2cd67fb045a4d6fe9e327da455c1b4cf7814f..3c9b4f060aee5aa3b0d7247681709d996bb4814c 100644
--- a/toolkit/modules/LightweightThemeConsumer.jsm
+++ b/toolkit/modules/LightweightThemeConsumer.jsm
@@ -6,9 +6,9 @@ var EXPORTED_SYMBOLS = ["LightweightThemeConsumer"];
 
 const { Services } = ChromeUtils.import("resource://gre/modules/Services.jsm");
 
-const DEFAULT_THEME_ID = "default-theme@mozilla.org";
-const LIGHT_THEME_ID = "firefox-compact-light@mozilla.org";
-const DARK_THEME_ID = "firefox-compact-dark@mozilla.org";
+const DEFAULT_THEME_ID = null;
+const LIGHT_THEME_ID = Services.builtInThemes.LIGHT_THEME_ID;
+const DARK_THEME_ID = Services.builtInThemes.DARK_THEME_ID;
 
 ChromeUtils.defineModuleGetter(
   this,
@@ -293,8 +293,8 @@ LightweightThemeConsumer.prototype = {
     root.toggleAttribute(
       "lwtheme-mozlightdark",
       theme.id == DEFAULT_THEME_ID ||
-        theme.id == LIGHT_THEME_ID ||
-        theme.id == DARK_THEME_ID
+      theme.id == LIGHT_THEME_ID ||
+      theme.id == DARK_THEME_ID
     );
     this._setExperiment(active, themeData.experiment, theme.experimental);
     _setImage(root, active, "--lwt-header-image", theme.headerURL);
