diff --git a/browser/themes/addons/jar.mn b/browser/themes/addons/jar.mn
index 5aa122b2ab38838daf4d08782f7651c107ef8b4b..680d9e6a6cefad6491c9b159077b138fa523f639 100644
--- a/browser/themes/addons/jar.mn
+++ b/browser/themes/addons/jar.mn
@@ -5,13 +5,10 @@
 browser.jar:
 %  resource builtin-themes %content/builtin-themes/
 
-  content/builtin-themes/alpenglow                 (alpenglow/*.svg)
-  content/builtin-themes/alpenglow/manifest.json   (alpenglow/manifest.json)
-
   content/builtin-themes/dark                      (dark/*.svg)
   content/builtin-themes/dark                      (dark/*.css)
   content/builtin-themes/dark/manifest.json        (dark/manifest.json)
 
   content/builtin-themes/light                     (light/*.svg)
   content/builtin-themes/light                     (light/*.css)
-  content/builtin-themes/light/manifest.json       (light/manifest.json)
+  content/builtin-themes/light/manifest.json       (light/manifest.json)
\ No newline at end of file
