diff --git a/browser/components/about/components.conf b/browser/components/about/components.conf
index 0c9597ff9fb4b2b01385511f22d37640cb7942ac..f4f9f5ad85e3c096cf1094aa3a32f4d63665a4b8 100644
--- a/browser/components/about/components.conf
+++ b/browser/components/about/components.conf
@@ -12,7 +12,8 @@ pages = [
     'home',
     'logins',
     'loginsimportreport',
-    'newtab',
+# This has been removed to dot/about/components.conf
+#   'newtab',
     'ion',
     'pocket-home',
     'pocket-saved',
