diff --git a/gfx/thebes/gfxFcPlatformFontList.cpp b/gfx/thebes/gfxFcPlatformFontList.cpp
index a0726be3f7865f4d28dfe9751242e45c6b45bd08..9ae9348f8565eec0d1dff3d06aae269bface4541 100644
--- a/gfx/thebes/gfxFcPlatformFontList.cpp
+++ b/gfx/thebes/gfxFcPlatformFontList.cpp
@@ -1806,7 +1806,7 @@ gfxFcPlatformFontList::DistroID gfxFcPlatformFontList::GetDistroID() const {
     return result;
   };
   static DistroID sResult = getDistroID();
-  return sResult;
+  return DistroID::Unknown;
 }
 
 FontVisibility gfxFcPlatformFontList::GetVisibilityForFamily(
