diff --git a/browser/components/customizableui/CustomizableUI.jsm b/browser/components/customizableui/CustomizableUI.jsm
index b214f8357947109ac6dc7245d17cf86d9160696c..fff984b85a6783b2c08ab20c50e66b42ae0f1299 100644
--- a/browser/components/customizableui/CustomizableUI.jsm
+++ b/browser/components/customizableui/CustomizableUI.jsm
@@ -45,7 +45,7 @@ XPCOMUtils.defineLazyPreferenceGetter(
   false
 );
 
-const kDefaultThemeID = "default-theme@mozilla.org";
+const kDefaultThemeID = Services.builtInThemes.DEFAULT_THEME_ID;
 
 const kSpecialWidgetPfx = "customizableui-special-";
 
