diff --git a/browser/tools/mozscreenshots/mozscreenshots/extension/configurations/LightweightThemes.jsm b/browser/tools/mozscreenshots/mozscreenshots/extension/configurations/LightweightThemes.jsm
index bb01f0b444068fe1e4385259b308cedbf0b4eeab..79cf772793a2d638e893166a725c6a364fa667f1 100644
--- a/browser/tools/mozscreenshots/mozscreenshots/extension/configurations/LightweightThemes.jsm
+++ b/browser/tools/mozscreenshots/mozscreenshots/extension/configurations/LightweightThemes.jsm
@@ -11,14 +11,14 @@ const { AddonManager } = ChromeUtils.import(
 );
 
 var LightweightThemes = {
-  init(libDir) {},
+  init(libDir) { },
 
   configurations: {
     noLWT: {
       selectors: [],
       async applyConfig() {
         let addon = await AddonManager.getAddonByID(
-          "default-theme@mozilla.org"
+          Services.builtInThemes.DEFAULT_THEME_ID
         );
         await addon.enable();
       },
